// @klaytn/web3rpc/dist/model/BlockWithConsensusInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The BlockWithConsensusInfo model module.
 * @module model/BlockWithConsensusInfo
 * @version 0.9.8
 */
var BlockWithConsensusInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BlockWithConsensusInfo</code>.
   * @alias module:model/BlockWithConsensusInfo
   * @param blockScore {String} Former difficulty. Always 1 in the BFT consensus engine
   * @param totalBlockScore {String} Integer of the total blockScore of the chain until this block.
   * @param committee {Array.<String>} Array of addresses of committee members of this block. The committee is a subset of validators participated in the consensus protocol for this block.
   * @param gasUsed {String} The total used gas by all transactions in this block.
   * @param hash {String} Hash of the block. null when it is pending block.
   * @param number {String} The block number. null when it is pending block.
   * @param parentHash {String} Hash of the parent block.
   * @param proposer {String} Address of the proposer.
   * @param receiptsRoot {String} The root of the receipts trie of the block.
   * @param size {String} Integer the size of this block in bytes.
   * @param stateRoot {String} The root of the final state trie of the block.
   * @param timestamp {String} The Unix timestamp for when the block was collated.
   * @param timestampFoS {String} The fraction of a second of the timestamp for when the block was collated.
   * @param transactions {Array.<Object>} Array of transaction objects.
   * @param transactionsRoot {String} The root of the transaction trie of the block.
   * @param baseFeePerGas {String} The base fee per gas. It has a meaningful value when EthTxTypeCompatible and Magma hardforks are activated.
   */
  function BlockWithConsensusInfo(blockScore, totalBlockScore, committee, gasUsed, hash, number, parentHash, proposer, receiptsRoot, size, stateRoot, timestamp, timestampFoS, transactions, transactionsRoot, baseFeePerGas) {
    _classCallCheck(this, BlockWithConsensusInfo);
    BlockWithConsensusInfo.initialize(this, blockScore, totalBlockScore, committee, gasUsed, hash, number, parentHash, proposer, receiptsRoot, size, stateRoot, timestamp, timestampFoS, transactions, transactionsRoot, baseFeePerGas);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(BlockWithConsensusInfo, null, [{
    key: "initialize",
    value: function initialize(obj, blockScore, totalBlockScore, committee, gasUsed, hash, number, parentHash, proposer, receiptsRoot, size, stateRoot, timestamp, timestampFoS, transactions, transactionsRoot, baseFeePerGas) {
      obj['blockScore'] = blockScore;
      obj['totalBlockScore'] = totalBlockScore;
      obj['committee'] = committee;
      obj['gasUsed'] = gasUsed;
      obj['hash'] = hash;
      obj['number'] = number;
      obj['parentHash'] = parentHash;
      obj['proposer'] = proposer;
      obj['receiptsRoot'] = receiptsRoot;
      obj['size'] = size;
      obj['stateRoot'] = stateRoot;
      obj['timestamp'] = timestamp;
      obj['timestampFoS'] = timestampFoS;
      obj['transactions'] = transactions;
      obj['transactionsRoot'] = transactionsRoot;
      obj['baseFeePerGas'] = baseFeePerGas;
    }

    /**
     * Constructs a <code>BlockWithConsensusInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BlockWithConsensusInfo} obj Optional instance to populate.
     * @return {module:model/BlockWithConsensusInfo} The populated <code>BlockWithConsensusInfo</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BlockWithConsensusInfo();
        if (data.hasOwnProperty('blockScore')) {
          obj['blockScore'] = _ApiClient["default"].convertToType(data['blockScore'], 'String');
        }
        if (data.hasOwnProperty('totalBlockScore')) {
          obj['totalBlockScore'] = _ApiClient["default"].convertToType(data['totalBlockScore'], 'String');
        }
        if (data.hasOwnProperty('committee')) {
          obj['committee'] = _ApiClient["default"].convertToType(data['committee'], ['String']);
        }
        if (data.hasOwnProperty('gasUsed')) {
          obj['gasUsed'] = _ApiClient["default"].convertToType(data['gasUsed'], 'String');
        }
        if (data.hasOwnProperty('hash')) {
          obj['hash'] = _ApiClient["default"].convertToType(data['hash'], 'String');
        }
        if (data.hasOwnProperty('number')) {
          obj['number'] = _ApiClient["default"].convertToType(data['number'], 'String');
        }
        if (data.hasOwnProperty('parentHash')) {
          obj['parentHash'] = _ApiClient["default"].convertToType(data['parentHash'], 'String');
        }
        if (data.hasOwnProperty('proposer')) {
          obj['proposer'] = _ApiClient["default"].convertToType(data['proposer'], 'String');
        }
        if (data.hasOwnProperty('receiptsRoot')) {
          obj['receiptsRoot'] = _ApiClient["default"].convertToType(data['receiptsRoot'], 'String');
        }
        if (data.hasOwnProperty('size')) {
          obj['size'] = _ApiClient["default"].convertToType(data['size'], 'String');
        }
        if (data.hasOwnProperty('stateRoot')) {
          obj['stateRoot'] = _ApiClient["default"].convertToType(data['stateRoot'], 'String');
        }
        if (data.hasOwnProperty('timestamp')) {
          obj['timestamp'] = _ApiClient["default"].convertToType(data['timestamp'], 'String');
        }
        if (data.hasOwnProperty('timestampFoS')) {
          obj['timestampFoS'] = _ApiClient["default"].convertToType(data['timestampFoS'], 'String');
        }
        if (data.hasOwnProperty('transactions')) {
          obj['transactions'] = _ApiClient["default"].convertToType(data['transactions'], [Object]);
        }
        if (data.hasOwnProperty('transactionsRoot')) {
          obj['transactionsRoot'] = _ApiClient["default"].convertToType(data['transactionsRoot'], 'String');
        }
        if (data.hasOwnProperty('baseFeePerGas')) {
          obj['baseFeePerGas'] = _ApiClient["default"].convertToType(data['baseFeePerGas'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>BlockWithConsensusInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>BlockWithConsensusInfo</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(BlockWithConsensusInfo.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['blockScore'] && !(typeof data['blockScore'] === 'string' || data['blockScore'] instanceof String)) {
        throw new Error("Expected the field `blockScore` to be a primitive type in the JSON string but got " + data['blockScore']);
      }
      // ensure the json data is a string
      if (data['totalBlockScore'] && !(typeof data['totalBlockScore'] === 'string' || data['totalBlockScore'] instanceof String)) {
        throw new Error("Expected the field `totalBlockScore` to be a primitive type in the JSON string but got " + data['totalBlockScore']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['committee'])) {
        throw new Error("Expected the field `committee` to be an array in the JSON data but got " + data['committee']);
      }
      // ensure the json data is a string
      if (data['gasUsed'] && !(typeof data['gasUsed'] === 'string' || data['gasUsed'] instanceof String)) {
        throw new Error("Expected the field `gasUsed` to be a primitive type in the JSON string but got " + data['gasUsed']);
      }
      // ensure the json data is a string
      if (data['hash'] && !(typeof data['hash'] === 'string' || data['hash'] instanceof String)) {
        throw new Error("Expected the field `hash` to be a primitive type in the JSON string but got " + data['hash']);
      }
      // ensure the json data is a string
      if (data['number'] && !(typeof data['number'] === 'string' || data['number'] instanceof String)) {
        throw new Error("Expected the field `number` to be a primitive type in the JSON string but got " + data['number']);
      }
      // ensure the json data is a string
      if (data['parentHash'] && !(typeof data['parentHash'] === 'string' || data['parentHash'] instanceof String)) {
        throw new Error("Expected the field `parentHash` to be a primitive type in the JSON string but got " + data['parentHash']);
      }
      // ensure the json data is a string
      if (data['proposer'] && !(typeof data['proposer'] === 'string' || data['proposer'] instanceof String)) {
        throw new Error("Expected the field `proposer` to be a primitive type in the JSON string but got " + data['proposer']);
      }
      // ensure the json data is a string
      if (data['receiptsRoot'] && !(typeof data['receiptsRoot'] === 'string' || data['receiptsRoot'] instanceof String)) {
        throw new Error("Expected the field `receiptsRoot` to be a primitive type in the JSON string but got " + data['receiptsRoot']);
      }
      // ensure the json data is a string
      if (data['size'] && !(typeof data['size'] === 'string' || data['size'] instanceof String)) {
        throw new Error("Expected the field `size` to be a primitive type in the JSON string but got " + data['size']);
      }
      // ensure the json data is a string
      if (data['stateRoot'] && !(typeof data['stateRoot'] === 'string' || data['stateRoot'] instanceof String)) {
        throw new Error("Expected the field `stateRoot` to be a primitive type in the JSON string but got " + data['stateRoot']);
      }
      // ensure the json data is a string
      if (data['timestamp'] && !(typeof data['timestamp'] === 'string' || data['timestamp'] instanceof String)) {
        throw new Error("Expected the field `timestamp` to be a primitive type in the JSON string but got " + data['timestamp']);
      }
      // ensure the json data is a string
      if (data['timestampFoS'] && !(typeof data['timestampFoS'] === 'string' || data['timestampFoS'] instanceof String)) {
        throw new Error("Expected the field `timestampFoS` to be a primitive type in the JSON string but got " + data['timestampFoS']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['transactions'])) {
        throw new Error("Expected the field `transactions` to be an array in the JSON data but got " + data['transactions']);
      }
      // ensure the json data is a string
      if (data['transactionsRoot'] && !(typeof data['transactionsRoot'] === 'string' || data['transactionsRoot'] instanceof String)) {
        throw new Error("Expected the field `transactionsRoot` to be a primitive type in the JSON string but got " + data['transactionsRoot']);
      }
      // ensure the json data is a string
      if (data['baseFeePerGas'] && !(typeof data['baseFeePerGas'] === 'string' || data['baseFeePerGas'] instanceof String)) {
        throw new Error("Expected the field `baseFeePerGas` to be a primitive type in the JSON string but got " + data['baseFeePerGas']);
      }
      return true;
    }
  }]);
  return BlockWithConsensusInfo;
}();
BlockWithConsensusInfo.RequiredProperties = ["blockScore", "totalBlockScore", "committee", "gasUsed", "hash", "number", "parentHash", "proposer", "receiptsRoot", "size", "stateRoot", "timestamp", "timestampFoS", "transactions", "transactionsRoot", "baseFeePerGas"];

/**
 * Former difficulty. Always 1 in the BFT consensus engine
 * @member {String} blockScore
 */
BlockWithConsensusInfo.prototype['blockScore'] = undefined;

/**
 * Integer of the total blockScore of the chain until this block.
 * @member {String} totalBlockScore
 */
BlockWithConsensusInfo.prototype['totalBlockScore'] = undefined;

/**
 * Array of addresses of committee members of this block. The committee is a subset of validators participated in the consensus protocol for this block.
 * @member {Array.<String>} committee
 */
BlockWithConsensusInfo.prototype['committee'] = undefined;

/**
 * The total used gas by all transactions in this block.
 * @member {String} gasUsed
 */
BlockWithConsensusInfo.prototype['gasUsed'] = undefined;

/**
 * Hash of the block. null when it is pending block.
 * @member {String} hash
 */
BlockWithConsensusInfo.prototype['hash'] = undefined;

/**
 * The block number. null when it is pending block.
 * @member {String} number
 */
BlockWithConsensusInfo.prototype['number'] = undefined;

/**
 * Hash of the parent block.
 * @member {String} parentHash
 */
BlockWithConsensusInfo.prototype['parentHash'] = undefined;

/**
 * Address of the proposer.
 * @member {String} proposer
 */
BlockWithConsensusInfo.prototype['proposer'] = undefined;

/**
 * The root of the receipts trie of the block.
 * @member {String} receiptsRoot
 */
BlockWithConsensusInfo.prototype['receiptsRoot'] = undefined;

/**
 * Integer the size of this block in bytes.
 * @member {String} size
 */
BlockWithConsensusInfo.prototype['size'] = undefined;

/**
 * The root of the final state trie of the block.
 * @member {String} stateRoot
 */
BlockWithConsensusInfo.prototype['stateRoot'] = undefined;

/**
 * The Unix timestamp for when the block was collated.
 * @member {String} timestamp
 */
BlockWithConsensusInfo.prototype['timestamp'] = undefined;

/**
 * The fraction of a second of the timestamp for when the block was collated.
 * @member {String} timestampFoS
 */
BlockWithConsensusInfo.prototype['timestampFoS'] = undefined;

/**
 * Array of transaction objects.
 * @member {Array.<Object>} transactions
 */
BlockWithConsensusInfo.prototype['transactions'] = undefined;

/**
 * The root of the transaction trie of the block.
 * @member {String} transactionsRoot
 */
BlockWithConsensusInfo.prototype['transactionsRoot'] = undefined;

/**
 * The base fee per gas. It has a meaningful value when EthTxTypeCompatible and Magma hardforks are activated.
 * @member {String} baseFeePerGas
 */
BlockWithConsensusInfo.prototype['baseFeePerGas'] = undefined;
var _default = exports["default"] = BlockWithConsensusInfo;
