// @klaytn/web3rpc/dist/model/ChainID200Response.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ChainID200ResponseAllOf = _interopRequireDefault(require("./ChainID200ResponseAllOf"));
var _ErrorMember = _interopRequireDefault(require("./ErrorMember"));
var _JsonRpcResponse = _interopRequireDefault(require("./JsonRpcResponse"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ChainID200Response model module.
 * @module model/ChainID200Response
 * @version 0.9.8
 */
var ChainID200Response = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ChainID200Response</code>.
   * @alias module:model/ChainID200Response
   * @implements module:model/JsonRpcResponse
   * @implements module:model/ChainID200ResponseAllOf
   * @param jsonrpc {String} 
   * @param id {Number} 
   */
  function ChainID200Response(jsonrpc, id) {
    _classCallCheck(this, ChainID200Response);
    _JsonRpcResponse["default"].initialize(this, jsonrpc, id);
    _ChainID200ResponseAllOf["default"].initialize(this);
    ChainID200Response.initialize(this, jsonrpc, id);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ChainID200Response, null, [{
    key: "initialize",
    value: function initialize(obj, jsonrpc, id) {
      obj['jsonrpc'] = jsonrpc || '2.0';
      obj['id'] = id;
    }

    /**
     * Constructs a <code>ChainID200Response</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ChainID200Response} obj Optional instance to populate.
     * @return {module:model/ChainID200Response} The populated <code>ChainID200Response</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ChainID200Response();
        _JsonRpcResponse["default"].constructFromObject(data, obj);
        _ChainID200ResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('jsonrpc')) {
          obj['jsonrpc'] = _ApiClient["default"].convertToType(data['jsonrpc'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('error')) {
          obj['error'] = _ErrorMember["default"].constructFromObject(data['error']);
        }
        if (data.hasOwnProperty('result')) {
          obj['result'] = _ApiClient["default"].convertToType(data['result'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ChainID200Response</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ChainID200Response</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ChainID200Response.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['jsonrpc'] && !(typeof data['jsonrpc'] === 'string' || data['jsonrpc'] instanceof String)) {
        throw new Error("Expected the field `jsonrpc` to be a primitive type in the JSON string but got " + data['jsonrpc']);
      }
      // validate the optional field `error`
      if (data['error']) {
        // data not null
        _ErrorMember["default"].validateJSON(data['error']);
      }
      // ensure the json data is a string
      if (data['result'] && !(typeof data['result'] === 'string' || data['result'] instanceof String)) {
        throw new Error("Expected the field `result` to be a primitive type in the JSON string but got " + data['result']);
      }
      return true;
    }
  }]);
  return ChainID200Response;
}();
ChainID200Response.RequiredProperties = ["jsonrpc", "id"];

/**
 * @member {String} jsonrpc
 * @default '2.0'
 */
ChainID200Response.prototype['jsonrpc'] = '2.0';

/**
 * @member {Number} id
 */
ChainID200Response.prototype['id'] = undefined;

/**
 * @member {module:model/ErrorMember} error
 */
ChainID200Response.prototype['error'] = undefined;

/**
 * Integer of the chain ID of the chain.
 * @member {String} result
 */
ChainID200Response.prototype['result'] = undefined;

// Implement JsonRpcResponse interface:
/**
 * @member {String} jsonrpc
 * @default '2.0'
 */
_JsonRpcResponse["default"].prototype['jsonrpc'] = '2.0';
/**
 * @member {Number} id
 */
_JsonRpcResponse["default"].prototype['id'] = undefined;
/**
 * @member {module:model/ErrorMember} error
 */
_JsonRpcResponse["default"].prototype['error'] = undefined;
/**
 * @member {Object} result
 */
_JsonRpcResponse["default"].prototype['result'] = undefined;
// Implement ChainID200ResponseAllOf interface:
/**
 * Integer of the chain ID of the chain.
 * @member {String} result
 */
_ChainID200ResponseAllOf["default"].prototype['result'] = undefined;
var _default = exports["default"] = ChainID200Response;
