// @klaytn/web3rpc/dist/model/DebugTraceBlockRespResultInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The DebugTraceBlockRespResultInner model module.
 * @module model/DebugTraceBlockRespResultInner
 * @version 0.9.8
 */
var DebugTraceBlockRespResultInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DebugTraceBlockRespResultInner</code>.
   * @alias module:model/DebugTraceBlockRespResultInner
   */
  function DebugTraceBlockRespResultInner() {
    _classCallCheck(this, DebugTraceBlockRespResultInner);
    DebugTraceBlockRespResultInner.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(DebugTraceBlockRespResultInner, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>DebugTraceBlockRespResultInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DebugTraceBlockRespResultInner} obj Optional instance to populate.
     * @return {module:model/DebugTraceBlockRespResultInner} The populated <code>DebugTraceBlockRespResultInner</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DebugTraceBlockRespResultInner();
        if (data.hasOwnProperty('failed')) {
          obj['failed'] = _ApiClient["default"].convertToType(data['failed'], 'Boolean');
        }
        if (data.hasOwnProperty('gas')) {
          obj['gas'] = _ApiClient["default"].convertToType(data['gas'], 'Number');
        }
        if (data.hasOwnProperty('returnValue')) {
          obj['returnValue'] = _ApiClient["default"].convertToType(data['returnValue'], 'String');
        }
        if (data.hasOwnProperty('structLogs')) {
          obj['structLogs'] = _ApiClient["default"].convertToType(data['structLogs'], [Object]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>DebugTraceBlockRespResultInner</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DebugTraceBlockRespResultInner</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['returnValue'] && !(typeof data['returnValue'] === 'string' || data['returnValue'] instanceof String)) {
        throw new Error("Expected the field `returnValue` to be a primitive type in the JSON string but got " + data['returnValue']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['structLogs'])) {
        throw new Error("Expected the field `structLogs` to be an array in the JSON data but got " + data['structLogs']);
      }
      return true;
    }
  }]);
  return DebugTraceBlockRespResultInner;
}();
/**
 * @member {Boolean} failed
 */
DebugTraceBlockRespResultInner.prototype['failed'] = undefined;

/**
 * @member {Number} gas
 */
DebugTraceBlockRespResultInner.prototype['gas'] = undefined;

/**
 * @member {String} returnValue
 */
DebugTraceBlockRespResultInner.prototype['returnValue'] = undefined;

/**
 * @member {Array.<Object>} structLogs
 */
DebugTraceBlockRespResultInner.prototype['structLogs'] = undefined;
var _default = exports["default"] = DebugTraceBlockRespResultInner;
