// @klaytn/web3rpc/dist/model/GcStatsRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GcStatsRespResult model module.
 * @module model/GcStatsRespResult
 * @version 0.9.8
 */
var GcStatsRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GcStatsRespResult</code>.
   * GC statistics
   * @alias module:model/GcStatsRespResult
   * @param lastGC {Date} time of last collection
   * @param numGC {Number} number of garbage collections
   * @param pause {Array.<Number>} pause history, most recent first
   * @param pauseEnd {Array.<Date>} pause end times history, most recent first
   * @param pauseQuantiles {String} 
   * @param pauseTotal {Number} total pause for all collections
   */
  function GcStatsRespResult(lastGC, numGC, pause, pauseEnd, pauseQuantiles, pauseTotal) {
    _classCallCheck(this, GcStatsRespResult);
    GcStatsRespResult.initialize(this, lastGC, numGC, pause, pauseEnd, pauseQuantiles, pauseTotal);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(GcStatsRespResult, null, [{
    key: "initialize",
    value: function initialize(obj, lastGC, numGC, pause, pauseEnd, pauseQuantiles, pauseTotal) {
      obj['LastGC'] = lastGC;
      obj['NumGC'] = numGC;
      obj['Pause'] = pause;
      obj['PauseEnd'] = pauseEnd;
      obj['PauseQuantiles'] = pauseQuantiles;
      obj['PauseTotal'] = pauseTotal;
    }

    /**
     * Constructs a <code>GcStatsRespResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GcStatsRespResult} obj Optional instance to populate.
     * @return {module:model/GcStatsRespResult} The populated <code>GcStatsRespResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GcStatsRespResult();
        if (data.hasOwnProperty('LastGC')) {
          obj['LastGC'] = _ApiClient["default"].convertToType(data['LastGC'], 'Date');
        }
        if (data.hasOwnProperty('NumGC')) {
          obj['NumGC'] = _ApiClient["default"].convertToType(data['NumGC'], 'Number');
        }
        if (data.hasOwnProperty('Pause')) {
          obj['Pause'] = _ApiClient["default"].convertToType(data['Pause'], ['Number']);
        }
        if (data.hasOwnProperty('PauseEnd')) {
          obj['PauseEnd'] = _ApiClient["default"].convertToType(data['PauseEnd'], ['Date']);
        }
        if (data.hasOwnProperty('PauseQuantiles')) {
          obj['PauseQuantiles'] = _ApiClient["default"].convertToType(data['PauseQuantiles'], 'String');
        }
        if (data.hasOwnProperty('PauseTotal')) {
          obj['PauseTotal'] = _ApiClient["default"].convertToType(data['PauseTotal'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>GcStatsRespResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GcStatsRespResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GcStatsRespResult.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is an array
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (!Array.isArray(data['Pause'])) {
        throw new Error("Expected the field `Pause` to be an array in the JSON data but got " + data['Pause']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['PauseEnd'])) {
        throw new Error("Expected the field `PauseEnd` to be an array in the JSON data but got " + data['PauseEnd']);
      }
      // ensure the json data is a string
      if (data['PauseQuantiles'] && !(typeof data['PauseQuantiles'] === 'string' || data['PauseQuantiles'] instanceof String)) {
        throw new Error("Expected the field `PauseQuantiles` to be a primitive type in the JSON string but got " + data['PauseQuantiles']);
      }
      return true;
    }
  }]);
  return GcStatsRespResult;
}();
GcStatsRespResult.RequiredProperties = ["LastGC", "NumGC", "Pause", "PauseEnd", "PauseQuantiles", "PauseTotal"];

/**
 * time of last collection
 * @member {Date} LastGC
 */
GcStatsRespResult.prototype['LastGC'] = undefined;

/**
 * number of garbage collections
 * @member {Number} NumGC
 */
GcStatsRespResult.prototype['NumGC'] = undefined;

/**
 * pause history, most recent first
 * @member {Array.<Number>} Pause
 */
GcStatsRespResult.prototype['Pause'] = undefined;

/**
 * pause end times history, most recent first
 * @member {Array.<Date>} PauseEnd
 */
GcStatsRespResult.prototype['PauseEnd'] = undefined;

/**
 * @member {String} PauseQuantiles
 */
GcStatsRespResult.prototype['PauseQuantiles'] = undefined;

/**
 * total pause for all collections
 * @member {Number} PauseTotal
 */
GcStatsRespResult.prototype['PauseTotal'] = undefined;
var _default = exports["default"] = GcStatsRespResult;
