// @klaytn/web3rpc/dist/model/GetRewardsAccumulatedRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _GetRewardsAccumulatedRequestAllOf = _interopRequireDefault(require("./GetRewardsAccumulatedRequestAllOf"));
var _GetRewardsAccumulatedRequestAllOfParamsInner = _interopRequireDefault(require("./GetRewardsAccumulatedRequestAllOfParamsInner"));
var _JsonRpcRequest = _interopRequireDefault(require("./JsonRpcRequest"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetRewardsAccumulatedRequest model module.
 * @module model/GetRewardsAccumulatedRequest
 * @version 0.9.8
 */
var GetRewardsAccumulatedRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRewardsAccumulatedRequest</code>.
   * @alias module:model/GetRewardsAccumulatedRequest
   * @implements module:model/JsonRpcRequest
   * @implements module:model/GetRewardsAccumulatedRequestAllOf
   * @param method {String} 
   * @param id {Number} Request ID, Random or Auto incrementation
   * @param jsonrpc {String} JSON-RPC Version (2.0)
   * @param params {Array.<module:model/GetRewardsAccumulatedRequestAllOfParamsInner>} Integer of a block number.
   */
  function GetRewardsAccumulatedRequest(method, id, jsonrpc, params) {
    _classCallCheck(this, GetRewardsAccumulatedRequest);
    _JsonRpcRequest["default"].initialize(this, method, id, jsonrpc);
    _GetRewardsAccumulatedRequestAllOf["default"].initialize(this, method, params);
    GetRewardsAccumulatedRequest.initialize(this, method, id, jsonrpc, params);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(GetRewardsAccumulatedRequest, null, [{
    key: "initialize",
    value: function initialize(obj, method, id, jsonrpc, params) {
      obj['method'] = method || 'governance_getRewardsAccumulated';
      obj['id'] = id || 1;
      obj['jsonrpc'] = jsonrpc || '2.0';
      obj['params'] = params;
    }

    /**
     * Constructs a <code>GetRewardsAccumulatedRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRewardsAccumulatedRequest} obj Optional instance to populate.
     * @return {module:model/GetRewardsAccumulatedRequest} The populated <code>GetRewardsAccumulatedRequest</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRewardsAccumulatedRequest();
        _JsonRpcRequest["default"].constructFromObject(data, obj);
        _GetRewardsAccumulatedRequestAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('method')) {
          obj['method'] = _ApiClient["default"].convertToType(data['method'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('jsonrpc')) {
          obj['jsonrpc'] = _ApiClient["default"].convertToType(data['jsonrpc'], 'String');
        }
        if (data.hasOwnProperty('params')) {
          obj['params'] = _ApiClient["default"].convertToType(data['params'], [_GetRewardsAccumulatedRequestAllOfParamsInner["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>GetRewardsAccumulatedRequest</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetRewardsAccumulatedRequest</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GetRewardsAccumulatedRequest.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['method'] && !(typeof data['method'] === 'string' || data['method'] instanceof String)) {
        throw new Error("Expected the field `method` to be a primitive type in the JSON string but got " + data['method']);
      }
      // ensure the json data is a string
      if (data['jsonrpc'] && !(typeof data['jsonrpc'] === 'string' || data['jsonrpc'] instanceof String)) {
        throw new Error("Expected the field `jsonrpc` to be a primitive type in the JSON string but got " + data['jsonrpc']);
      }
      if (data['params']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['params'])) {
          throw new Error("Expected the field `params` to be an array in the JSON data but got " + data['params']);
        }
        // validate the optional field `params` (array)
        var _iterator2 = _createForOfIteratorHelper(data['params']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _GetRewardsAccumulatedRequestAllOfParamsInner["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      return true;
    }
  }]);
  return GetRewardsAccumulatedRequest;
}();
GetRewardsAccumulatedRequest.RequiredProperties = ["method", "id", "jsonrpc", "params"];

/**
 * @member {String} method
 * @default 'governance_getRewardsAccumulated'
 */
GetRewardsAccumulatedRequest.prototype['method'] = 'governance_getRewardsAccumulated';

/**
 * Request ID, Random or Auto incrementation
 * @member {Number} id
 * @default 1
 */
GetRewardsAccumulatedRequest.prototype['id'] = 1;

/**
 * JSON-RPC Version (2.0)
 * @member {String} jsonrpc
 * @default '2.0'
 */
GetRewardsAccumulatedRequest.prototype['jsonrpc'] = '2.0';

/**
 * Integer of a block number.
 * @member {Array.<module:model/GetRewardsAccumulatedRequestAllOfParamsInner>} params
 */
GetRewardsAccumulatedRequest.prototype['params'] = undefined;

// Implement JsonRpcRequest interface:
/**
 * Method name
 * @member {String} method
 */
_JsonRpcRequest["default"].prototype['method'] = undefined;
/**
 * Request ID, Random or Auto incrementation
 * @member {Number} id
 * @default 1
 */
_JsonRpcRequest["default"].prototype['id'] = 1;
/**
 * JSON-RPC Version (2.0)
 * @member {String} jsonrpc
 * @default '2.0'
 */
_JsonRpcRequest["default"].prototype['jsonrpc'] = '2.0';
/**
 * @member {Array.<Object>} params
 */
_JsonRpcRequest["default"].prototype['params'] = undefined;
// Implement GetRewardsAccumulatedRequestAllOf interface:
/**
 * @member {String} method
 * @default 'governance_getRewardsAccumulated'
 */
_GetRewardsAccumulatedRequestAllOf["default"].prototype['method'] = 'governance_getRewardsAccumulated';
/**
 * Integer of a block number.
 * @member {Array.<module:model/GetRewardsAccumulatedRequestAllOfParamsInner>} params
 */
_GetRewardsAccumulatedRequestAllOf["default"].prototype['params'] = undefined;
var _default = exports["default"] = GetRewardsAccumulatedRequest;
