// @klaytn/web3rpc/dist/model/GetRewardsAccumulated200ResponseAllOfResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetRewardsAccumulated200ResponseAllOfResult model module.
 * @module model/GetRewardsAccumulated200ResponseAllOfResult
 * @version 0.9.8
 */
var GetRewardsAccumulated200ResponseAllOfResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRewardsAccumulated200ResponseAllOfResult</code>.
   * Accumulated reward information in a given block range
   * @alias module:model/GetRewardsAccumulated200ResponseAllOfResult
   */
  function GetRewardsAccumulated200ResponseAllOfResult() {
    _classCallCheck(this, GetRewardsAccumulated200ResponseAllOfResult);
    GetRewardsAccumulated200ResponseAllOfResult.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(GetRewardsAccumulated200ResponseAllOfResult, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetRewardsAccumulated200ResponseAllOfResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRewardsAccumulated200ResponseAllOfResult} obj Optional instance to populate.
     * @return {module:model/GetRewardsAccumulated200ResponseAllOfResult} The populated <code>GetRewardsAccumulated200ResponseAllOfResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRewardsAccumulated200ResponseAllOfResult();
        if (data.hasOwnProperty('firstBlockTime')) {
          obj['firstBlockTime'] = _ApiClient["default"].convertToType(data['firstBlockTime'], 'String');
        }
        if (data.hasOwnProperty('lastBlockTime')) {
          obj['lastBlockTime'] = _ApiClient["default"].convertToType(data['lastBlockTime'], 'String');
        }
        if (data.hasOwnProperty('firstBlock')) {
          obj['firstBlock'] = _ApiClient["default"].convertToType(data['firstBlock'], 'Number');
        }
        if (data.hasOwnProperty('lastBlock')) {
          obj['lastBlock'] = _ApiClient["default"].convertToType(data['lastBlock'], 'Number');
        }
        if (data.hasOwnProperty('totalMinted')) {
          obj['totalMinted'] = _ApiClient["default"].convertToType(data['totalMinted'], 'Number');
        }
        if (data.hasOwnProperty('totalTxFee')) {
          obj['totalTxFee'] = _ApiClient["default"].convertToType(data['totalTxFee'], 'Number');
        }
        if (data.hasOwnProperty('totalBurntTxFee')) {
          obj['totalBurntTxFee'] = _ApiClient["default"].convertToType(data['totalBurntTxFee'], 'Number');
        }
        if (data.hasOwnProperty('totalProposerRewards')) {
          obj['totalProposerRewards'] = _ApiClient["default"].convertToType(data['totalProposerRewards'], 'Number');
        }
        if (data.hasOwnProperty('totalStakingRewards')) {
          obj['totalStakingRewards'] = _ApiClient["default"].convertToType(data['totalStakingRewards'], 'Number');
        }
        if (data.hasOwnProperty('totalKFFRewards')) {
          obj['totalKFFRewards'] = _ApiClient["default"].convertToType(data['totalKFFRewards'], 'Number');
        }
        if (data.hasOwnProperty('totalKCFRewards')) {
          obj['totalKCFRewards'] = _ApiClient["default"].convertToType(data['totalKCFRewards'], 'Number');
        }
        if (data.hasOwnProperty('rewards')) {
          obj['rewards'] = _ApiClient["default"].convertToType(data['rewards'], Object);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>GetRewardsAccumulated200ResponseAllOfResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetRewardsAccumulated200ResponseAllOfResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['firstBlockTime'] && !(typeof data['firstBlockTime'] === 'string' || data['firstBlockTime'] instanceof String)) {
        throw new Error("Expected the field `firstBlockTime` to be a primitive type in the JSON string but got " + data['firstBlockTime']);
      }
      // ensure the json data is a string
      if (data['lastBlockTime'] && !(typeof data['lastBlockTime'] === 'string' || data['lastBlockTime'] instanceof String)) {
        throw new Error("Expected the field `lastBlockTime` to be a primitive type in the JSON string but got " + data['lastBlockTime']);
      }
      return true;
    }
  }]);
  return GetRewardsAccumulated200ResponseAllOfResult;
}();
/**
 * @member {String} firstBlockTime
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['firstBlockTime'] = undefined;

/**
 * @member {String} lastBlockTime
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['lastBlockTime'] = undefined;

/**
 * @member {Number} firstBlock
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['firstBlock'] = undefined;

/**
 * @member {Number} lastBlock
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['lastBlock'] = undefined;

/**
 * @member {Number} totalMinted
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['totalMinted'] = undefined;

/**
 * @member {Number} totalTxFee
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['totalTxFee'] = undefined;

/**
 * @member {Number} totalBurntTxFee
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['totalBurntTxFee'] = undefined;

/**
 * @member {Number} totalProposerRewards
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['totalProposerRewards'] = undefined;

/**
 * @member {Number} totalStakingRewards
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['totalStakingRewards'] = undefined;

/**
 * @member {Number} totalKFFRewards
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['totalKFFRewards'] = undefined;

/**
 * @member {Number} totalKCFRewards
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['totalKCFRewards'] = undefined;

/**
 * @member {Object} rewards
 */
GetRewardsAccumulated200ResponseAllOfResult.prototype['rewards'] = undefined;
var _default = exports["default"] = GetRewardsAccumulated200ResponseAllOfResult;
