// @klaytn/web3rpc/dist/model/KlayHeaderObject.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayHeaderObject model module.
 * @module model/KlayHeaderObject
 * @version 0.9.8
 */
var KlayHeaderObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayHeaderObject</code>.
   * The block header object.
   * @alias module:model/KlayHeaderObject
   * @param hash {String} Hash of the block. null when it is pending block.
   * @param parentHash {String} Hash of the parent block.
   * @param transactionsRoot {String} The root of the transaction trie of the block
   * @param stateRoot {String} The root of the final state trie of the block.
   * @param receiptsRoot {String} The root of the receipts trie of the block.
   * @param reward {String} The address of the beneficiary to whom the block rewards were given.
   * @param blockScore {String} Former difficulty. Always 1 in the BFT consensus engine
   * @param extraData {String} The \"extra data\" field of this block.
   * @param gasUsed {String} The total used gas by all transactions in this block.
   * @param timestamp {String} The Unix timestamp for when the block was collated.
   * @param timestampFoS {String} The fraction of a second of the timestamp for when the block was collated.
   * @param governanceData {String} RLP encoded governance configuration
   * @param voteData {String} RLP encoded governance vote of the proposer
   * @param baseFeePerGas {String} The base fee per gas. It has a meaningful value when EthTxTypeCompatible and Magma hardforks are activated.
   */
  function KlayHeaderObject(hash, parentHash, transactionsRoot, stateRoot, receiptsRoot, reward, blockScore, extraData, gasUsed, timestamp, timestampFoS, governanceData, voteData, baseFeePerGas) {
    _classCallCheck(this, KlayHeaderObject);
    KlayHeaderObject.initialize(this, hash, parentHash, transactionsRoot, stateRoot, receiptsRoot, reward, blockScore, extraData, gasUsed, timestamp, timestampFoS, governanceData, voteData, baseFeePerGas);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayHeaderObject, null, [{
    key: "initialize",
    value: function initialize(obj, hash, parentHash, transactionsRoot, stateRoot, receiptsRoot, reward, blockScore, extraData, gasUsed, timestamp, timestampFoS, governanceData, voteData, baseFeePerGas) {
      obj['hash'] = hash;
      obj['parentHash'] = parentHash;
      obj['transactionsRoot'] = transactionsRoot;
      obj['stateRoot'] = stateRoot;
      obj['receiptsRoot'] = receiptsRoot;
      obj['reward'] = reward;
      obj['blockScore'] = blockScore;
      obj['extraData'] = extraData;
      obj['gasUsed'] = gasUsed;
      obj['timestamp'] = timestamp;
      obj['timestampFoS'] = timestampFoS;
      obj['governanceData'] = governanceData;
      obj['voteData'] = voteData;
      obj['baseFeePerGas'] = baseFeePerGas;
    }

    /**
     * Constructs a <code>KlayHeaderObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayHeaderObject} obj Optional instance to populate.
     * @return {module:model/KlayHeaderObject} The populated <code>KlayHeaderObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayHeaderObject();
        if (data.hasOwnProperty('number')) {
          obj['number'] = _ApiClient["default"].convertToType(data['number'], 'String');
        }
        if (data.hasOwnProperty('hash')) {
          obj['hash'] = _ApiClient["default"].convertToType(data['hash'], 'String');
        }
        if (data.hasOwnProperty('parentHash')) {
          obj['parentHash'] = _ApiClient["default"].convertToType(data['parentHash'], 'String');
        }
        if (data.hasOwnProperty('logsBloom')) {
          obj['logsBloom'] = _ApiClient["default"].convertToType(data['logsBloom'], 'String');
        }
        if (data.hasOwnProperty('transactionsRoot')) {
          obj['transactionsRoot'] = _ApiClient["default"].convertToType(data['transactionsRoot'], 'String');
        }
        if (data.hasOwnProperty('stateRoot')) {
          obj['stateRoot'] = _ApiClient["default"].convertToType(data['stateRoot'], 'String');
        }
        if (data.hasOwnProperty('receiptsRoot')) {
          obj['receiptsRoot'] = _ApiClient["default"].convertToType(data['receiptsRoot'], 'String');
        }
        if (data.hasOwnProperty('reward')) {
          obj['reward'] = _ApiClient["default"].convertToType(data['reward'], 'String');
        }
        if (data.hasOwnProperty('blockScore')) {
          obj['blockScore'] = _ApiClient["default"].convertToType(data['blockScore'], 'String');
        }
        if (data.hasOwnProperty('extraData')) {
          obj['extraData'] = _ApiClient["default"].convertToType(data['extraData'], 'String');
        }
        if (data.hasOwnProperty('gasUsed')) {
          obj['gasUsed'] = _ApiClient["default"].convertToType(data['gasUsed'], 'String');
        }
        if (data.hasOwnProperty('timestamp')) {
          obj['timestamp'] = _ApiClient["default"].convertToType(data['timestamp'], 'String');
        }
        if (data.hasOwnProperty('timestampFoS')) {
          obj['timestampFoS'] = _ApiClient["default"].convertToType(data['timestampFoS'], 'String');
        }
        if (data.hasOwnProperty('governanceData')) {
          obj['governanceData'] = _ApiClient["default"].convertToType(data['governanceData'], 'String');
        }
        if (data.hasOwnProperty('voteData')) {
          obj['voteData'] = _ApiClient["default"].convertToType(data['voteData'], 'String');
        }
        if (data.hasOwnProperty('baseFeePerGas')) {
          obj['baseFeePerGas'] = _ApiClient["default"].convertToType(data['baseFeePerGas'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayHeaderObject</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayHeaderObject</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(KlayHeaderObject.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['number'] && !(typeof data['number'] === 'string' || data['number'] instanceof String)) {
        throw new Error("Expected the field `number` to be a primitive type in the JSON string but got " + data['number']);
      }
      // ensure the json data is a string
      if (data['hash'] && !(typeof data['hash'] === 'string' || data['hash'] instanceof String)) {
        throw new Error("Expected the field `hash` to be a primitive type in the JSON string but got " + data['hash']);
      }
      // ensure the json data is a string
      if (data['parentHash'] && !(typeof data['parentHash'] === 'string' || data['parentHash'] instanceof String)) {
        throw new Error("Expected the field `parentHash` to be a primitive type in the JSON string but got " + data['parentHash']);
      }
      // ensure the json data is a string
      if (data['logsBloom'] && !(typeof data['logsBloom'] === 'string' || data['logsBloom'] instanceof String)) {
        throw new Error("Expected the field `logsBloom` to be a primitive type in the JSON string but got " + data['logsBloom']);
      }
      // ensure the json data is a string
      if (data['transactionsRoot'] && !(typeof data['transactionsRoot'] === 'string' || data['transactionsRoot'] instanceof String)) {
        throw new Error("Expected the field `transactionsRoot` to be a primitive type in the JSON string but got " + data['transactionsRoot']);
      }
      // ensure the json data is a string
      if (data['stateRoot'] && !(typeof data['stateRoot'] === 'string' || data['stateRoot'] instanceof String)) {
        throw new Error("Expected the field `stateRoot` to be a primitive type in the JSON string but got " + data['stateRoot']);
      }
      // ensure the json data is a string
      if (data['receiptsRoot'] && !(typeof data['receiptsRoot'] === 'string' || data['receiptsRoot'] instanceof String)) {
        throw new Error("Expected the field `receiptsRoot` to be a primitive type in the JSON string but got " + data['receiptsRoot']);
      }
      // ensure the json data is a string
      if (data['reward'] && !(typeof data['reward'] === 'string' || data['reward'] instanceof String)) {
        throw new Error("Expected the field `reward` to be a primitive type in the JSON string but got " + data['reward']);
      }
      // ensure the json data is a string
      if (data['blockScore'] && !(typeof data['blockScore'] === 'string' || data['blockScore'] instanceof String)) {
        throw new Error("Expected the field `blockScore` to be a primitive type in the JSON string but got " + data['blockScore']);
      }
      // ensure the json data is a string
      if (data['extraData'] && !(typeof data['extraData'] === 'string' || data['extraData'] instanceof String)) {
        throw new Error("Expected the field `extraData` to be a primitive type in the JSON string but got " + data['extraData']);
      }
      // ensure the json data is a string
      if (data['gasUsed'] && !(typeof data['gasUsed'] === 'string' || data['gasUsed'] instanceof String)) {
        throw new Error("Expected the field `gasUsed` to be a primitive type in the JSON string but got " + data['gasUsed']);
      }
      // ensure the json data is a string
      if (data['timestamp'] && !(typeof data['timestamp'] === 'string' || data['timestamp'] instanceof String)) {
        throw new Error("Expected the field `timestamp` to be a primitive type in the JSON string but got " + data['timestamp']);
      }
      // ensure the json data is a string
      if (data['timestampFoS'] && !(typeof data['timestampFoS'] === 'string' || data['timestampFoS'] instanceof String)) {
        throw new Error("Expected the field `timestampFoS` to be a primitive type in the JSON string but got " + data['timestampFoS']);
      }
      // ensure the json data is a string
      if (data['governanceData'] && !(typeof data['governanceData'] === 'string' || data['governanceData'] instanceof String)) {
        throw new Error("Expected the field `governanceData` to be a primitive type in the JSON string but got " + data['governanceData']);
      }
      // ensure the json data is a string
      if (data['voteData'] && !(typeof data['voteData'] === 'string' || data['voteData'] instanceof String)) {
        throw new Error("Expected the field `voteData` to be a primitive type in the JSON string but got " + data['voteData']);
      }
      // ensure the json data is a string
      if (data['baseFeePerGas'] && !(typeof data['baseFeePerGas'] === 'string' || data['baseFeePerGas'] instanceof String)) {
        throw new Error("Expected the field `baseFeePerGas` to be a primitive type in the JSON string but got " + data['baseFeePerGas']);
      }
      return true;
    }
  }]);
  return KlayHeaderObject;
}();
KlayHeaderObject.RequiredProperties = ["hash", "parentHash", "transactionsRoot", "stateRoot", "receiptsRoot", "reward", "blockScore", "extraData", "gasUsed", "timestamp", "timestampFoS", "governanceData", "voteData", "baseFeePerGas"];

/**
 * The block number. null when its pending block.
 * @member {String} number
 */
KlayHeaderObject.prototype['number'] = undefined;

/**
 * Hash of the block. null when it is pending block.
 * @member {String} hash
 */
KlayHeaderObject.prototype['hash'] = undefined;

/**
 * Hash of the parent block.
 * @member {String} parentHash
 */
KlayHeaderObject.prototype['parentHash'] = undefined;

/**
 * The bloom filter for the logs of the block. null when it is pending block.
 * @member {String} logsBloom
 */
KlayHeaderObject.prototype['logsBloom'] = undefined;

/**
 * The root of the transaction trie of the block
 * @member {String} transactionsRoot
 */
KlayHeaderObject.prototype['transactionsRoot'] = undefined;

/**
 * The root of the final state trie of the block.
 * @member {String} stateRoot
 */
KlayHeaderObject.prototype['stateRoot'] = undefined;

/**
 * The root of the receipts trie of the block.
 * @member {String} receiptsRoot
 */
KlayHeaderObject.prototype['receiptsRoot'] = undefined;

/**
 * The address of the beneficiary to whom the block rewards were given.
 * @member {String} reward
 */
KlayHeaderObject.prototype['reward'] = undefined;

/**
 * Former difficulty. Always 1 in the BFT consensus engine
 * @member {String} blockScore
 */
KlayHeaderObject.prototype['blockScore'] = undefined;

/**
 * The \"extra data\" field of this block.
 * @member {String} extraData
 */
KlayHeaderObject.prototype['extraData'] = undefined;

/**
 * The total used gas by all transactions in this block.
 * @member {String} gasUsed
 */
KlayHeaderObject.prototype['gasUsed'] = undefined;

/**
 * The Unix timestamp for when the block was collated.
 * @member {String} timestamp
 */
KlayHeaderObject.prototype['timestamp'] = undefined;

/**
 * The fraction of a second of the timestamp for when the block was collated.
 * @member {String} timestampFoS
 */
KlayHeaderObject.prototype['timestampFoS'] = undefined;

/**
 * RLP encoded governance configuration
 * @member {String} governanceData
 */
KlayHeaderObject.prototype['governanceData'] = undefined;

/**
 * RLP encoded governance vote of the proposer
 * @member {String} voteData
 */
KlayHeaderObject.prototype['voteData'] = undefined;

/**
 * The base fee per gas. It has a meaningful value when EthTxTypeCompatible and Magma hardforks are activated.
 * @member {String} baseFeePerGas
 */
KlayHeaderObject.prototype['baseFeePerGas'] = undefined;
var _default = exports["default"] = KlayHeaderObject;
