// @klaytn/web3rpc/dist/model/KlayGetChainConfigRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _KlayGetChainConfigRespResultGovernance = _interopRequireDefault(require("./KlayGetChainConfigRespResultGovernance"));
var _KlayGetChainConfigRespResultIstanbul = _interopRequireDefault(require("./KlayGetChainConfigRespResultIstanbul"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetChainConfigRespResult model module.
 * @module model/KlayGetChainConfigRespResult
 * @version 0.9.8
 */
var KlayGetChainConfigRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetChainConfigRespResult</code>.
   * @alias module:model/KlayGetChainConfigRespResult
   */
  function KlayGetChainConfigRespResult() {
    _classCallCheck(this, KlayGetChainConfigRespResult);
    KlayGetChainConfigRespResult.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetChainConfigRespResult, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>KlayGetChainConfigRespResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetChainConfigRespResult} obj Optional instance to populate.
     * @return {module:model/KlayGetChainConfigRespResult} The populated <code>KlayGetChainConfigRespResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetChainConfigRespResult();
        if (data.hasOwnProperty('chainId')) {
          obj['chainId'] = _ApiClient["default"].convertToType(data['chainId'], 'Number');
        }
        if (data.hasOwnProperty('deriveShaImpl')) {
          obj['deriveShaImpl'] = _ApiClient["default"].convertToType(data['deriveShaImpl'], 'String');
        }
        if (data.hasOwnProperty('ethTxTypeCompatibleBlock')) {
          obj['ethTxTypeCompatibleBlock'] = _ApiClient["default"].convertToType(data['ethTxTypeCompatibleBlock'], 'Number');
        }
        if (data.hasOwnProperty('governance')) {
          obj['governance'] = _KlayGetChainConfigRespResultGovernance["default"].constructFromObject(data['governance']);
        }
        if (data.hasOwnProperty('istanbul')) {
          obj['istanbul'] = _KlayGetChainConfigRespResultIstanbul["default"].constructFromObject(data['istanbul']);
        }
        if (data.hasOwnProperty('istanbulCompatibleBlock')) {
          obj['istanbulCompatibleBlock'] = _ApiClient["default"].convertToType(data['istanbulCompatibleBlock'], 'Number');
        }
        if (data.hasOwnProperty('koreCompatibleBlock')) {
          obj['koreCompatibleBlock'] = _ApiClient["default"].convertToType(data['koreCompatibleBlock'], 'Number');
        }
        if (data.hasOwnProperty('londonCompatibleBlock')) {
          obj['londonCompatibleBlock'] = _ApiClient["default"].convertToType(data['londonCompatibleBlock'], 'Number');
        }
        if (data.hasOwnProperty('magmaCompatibleBlock')) {
          obj['magmaCompatibleBlock'] = _ApiClient["default"].convertToType(data['magmaCompatibleBlock'], 'Number');
        }
        if (data.hasOwnProperty('unitPrice')) {
          obj['unitPrice'] = _ApiClient["default"].convertToType(data['unitPrice'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetChainConfigRespResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetChainConfigRespResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['deriveShaImpl'] && !(typeof data['deriveShaImpl'] === 'string' || data['deriveShaImpl'] instanceof String)) {
        throw new Error("Expected the field `deriveShaImpl` to be a primitive type in the JSON string but got " + data['deriveShaImpl']);
      }
      // validate the optional field `governance`
      if (data['governance']) {
        // data not null
        _KlayGetChainConfigRespResultGovernance["default"].validateJSON(data['governance']);
      }
      // validate the optional field `istanbul`
      if (data['istanbul']) {
        // data not null
        _KlayGetChainConfigRespResultIstanbul["default"].validateJSON(data['istanbul']);
      }
      return true;
    }
  }]);
  return KlayGetChainConfigRespResult;
}();
/**
 * @member {Number} chainId
 */
KlayGetChainConfigRespResult.prototype['chainId'] = undefined;

/**
 * @member {String} deriveShaImpl
 */
KlayGetChainConfigRespResult.prototype['deriveShaImpl'] = undefined;

/**
 * @member {Number} ethTxTypeCompatibleBlock
 */
KlayGetChainConfigRespResult.prototype['ethTxTypeCompatibleBlock'] = undefined;

/**
 * @member {module:model/KlayGetChainConfigRespResultGovernance} governance
 */
KlayGetChainConfigRespResult.prototype['governance'] = undefined;

/**
 * @member {module:model/KlayGetChainConfigRespResultIstanbul} istanbul
 */
KlayGetChainConfigRespResult.prototype['istanbul'] = undefined;

/**
 * @member {Number} istanbulCompatibleBlock
 */
KlayGetChainConfigRespResult.prototype['istanbulCompatibleBlock'] = undefined;

/**
 * @member {Number} koreCompatibleBlock
 */
KlayGetChainConfigRespResult.prototype['koreCompatibleBlock'] = undefined;

/**
 * @member {Number} londonCompatibleBlock
 */
KlayGetChainConfigRespResult.prototype['londonCompatibleBlock'] = undefined;

/**
 * @member {Number} magmaCompatibleBlock
 */
KlayGetChainConfigRespResult.prototype['magmaCompatibleBlock'] = undefined;

/**
 * @member {Number} unitPrice
 */
KlayGetChainConfigRespResult.prototype['unitPrice'] = undefined;
var _default = exports["default"] = KlayGetChainConfigRespResult;
