// @klaytn/web3rpc/dist/model/KlayCreateAccessListRespResultAccessListInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayCreateAccessListRespResultAccessListInner model module.
 * @module model/KlayCreateAccessListRespResultAccessListInner
 * @version 0.9.8
 */
var KlayCreateAccessListRespResultAccessListInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayCreateAccessListRespResultAccessListInner</code>.
   * @alias module:model/KlayCreateAccessListRespResultAccessListInner
   * @param oldestBlock {String} Lowest number block of returned range.
   * @param reward {Array.<Array.<String>>} A two-dimensional array of effective priority fees per gas at the requested block percentiles.
   * @param baseFeePerGas {Array.<String>} An array of block base fees per gas. This includes the next block after the newest of the returned range, because this value can be derived from the newest block. Zeroes are returned for pre-EIP-1559 blocks.
   * @param gasUsedRatio {Array.<String>} An array of gasUsed/gasLimit in the block.
   */
  function KlayCreateAccessListRespResultAccessListInner(oldestBlock, reward, baseFeePerGas, gasUsedRatio) {
    _classCallCheck(this, KlayCreateAccessListRespResultAccessListInner);
    KlayCreateAccessListRespResultAccessListInner.initialize(this, oldestBlock, reward, baseFeePerGas, gasUsedRatio);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayCreateAccessListRespResultAccessListInner, null, [{
    key: "initialize",
    value: function initialize(obj, oldestBlock, reward, baseFeePerGas, gasUsedRatio) {
      obj['oldestBlock'] = oldestBlock;
      obj['reward'] = reward;
      obj['baseFeePerGas'] = baseFeePerGas;
      obj['gasUsedRatio'] = gasUsedRatio;
    }

    /**
     * Constructs a <code>KlayCreateAccessListRespResultAccessListInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayCreateAccessListRespResultAccessListInner} obj Optional instance to populate.
     * @return {module:model/KlayCreateAccessListRespResultAccessListInner} The populated <code>KlayCreateAccessListRespResultAccessListInner</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayCreateAccessListRespResultAccessListInner();
        if (data.hasOwnProperty('oldestBlock')) {
          obj['oldestBlock'] = _ApiClient["default"].convertToType(data['oldestBlock'], 'String');
        }
        if (data.hasOwnProperty('reward')) {
          obj['reward'] = _ApiClient["default"].convertToType(data['reward'], [['String']]);
        }
        if (data.hasOwnProperty('baseFeePerGas')) {
          obj['baseFeePerGas'] = _ApiClient["default"].convertToType(data['baseFeePerGas'], ['String']);
        }
        if (data.hasOwnProperty('gasUsedRatio')) {
          obj['gasUsedRatio'] = _ApiClient["default"].convertToType(data['gasUsedRatio'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayCreateAccessListRespResultAccessListInner</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayCreateAccessListRespResultAccessListInner</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(KlayCreateAccessListRespResultAccessListInner.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['oldestBlock'] && !(typeof data['oldestBlock'] === 'string' || data['oldestBlock'] instanceof String)) {
        throw new Error("Expected the field `oldestBlock` to be a primitive type in the JSON string but got " + data['oldestBlock']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['reward'])) {
        throw new Error("Expected the field `reward` to be an array in the JSON data but got " + data['reward']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['baseFeePerGas'])) {
        throw new Error("Expected the field `baseFeePerGas` to be an array in the JSON data but got " + data['baseFeePerGas']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['gasUsedRatio'])) {
        throw new Error("Expected the field `gasUsedRatio` to be an array in the JSON data but got " + data['gasUsedRatio']);
      }
      return true;
    }
  }]);
  return KlayCreateAccessListRespResultAccessListInner;
}();
KlayCreateAccessListRespResultAccessListInner.RequiredProperties = ["oldestBlock", "reward", "baseFeePerGas", "gasUsedRatio"];

/**
 * Lowest number block of returned range.
 * @member {String} oldestBlock
 */
KlayCreateAccessListRespResultAccessListInner.prototype['oldestBlock'] = undefined;

/**
 * A two-dimensional array of effective priority fees per gas at the requested block percentiles.
 * @member {Array.<Array.<String>>} reward
 */
KlayCreateAccessListRespResultAccessListInner.prototype['reward'] = undefined;

/**
 * An array of block base fees per gas. This includes the next block after the newest of the returned range, because this value can be derived from the newest block. Zeroes are returned for pre-EIP-1559 blocks.
 * @member {Array.<String>} baseFeePerGas
 */
KlayCreateAccessListRespResultAccessListInner.prototype['baseFeePerGas'] = undefined;

/**
 * An array of gasUsed/gasLimit in the block.
 * @member {Array.<String>} gasUsedRatio
 */
KlayCreateAccessListRespResultAccessListInner.prototype['gasUsedRatio'] = undefined;
var _default = exports["default"] = KlayCreateAccessListRespResultAccessListInner;
