// @klaytn/web3rpc/dist/model/KlaySyncingRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _SyncingObject = _interopRequireDefault(require("./SyncingObject"));
var _class;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlaySyncingRespResult model module.
 * @module model/KlaySyncingRespResult
 * @version 0.9.8
 */
var KlaySyncingRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlaySyncingRespResult</code>.
   * @alias module:model/KlaySyncingRespResult
   * @param {(module:model/Boolean|module:model/SyncingObject)} instance The actual instance to initialize KlaySyncingRespResult.
   */
  function KlaySyncingRespResult() {
    var instance = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : null;
    _classCallCheck(this, KlaySyncingRespResult);
    /**
     * Returns the JSON representation of the actual intance.
     * @return {string}
     */
    _defineProperty(this, "toJSON", function () {
      return this.getActualInstance();
    });
    if (instance === null) {
      this.actualInstance = null;
      return;
    }
    var match = 0;
    var errorMessages = [];
    try {
      if (typeof instance === "SyncingObject") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _SyncingObject["default"].validateJSON(instance); // throw an exception if no match
        // create SyncingObject from JS object
        this.actualInstance = _SyncingObject["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into SyncingObject
      errorMessages.push("Failed to construct SyncingObject: " + err);
    }
    try {
      this.actualInstance = instance;
      match++;
    } catch (err) {
      // json data failed to deserialize into Boolean
      errorMessages.push("Failed to construct Boolean: " + err);
    }
    if (match > 1) {
      throw new Error("Multiple matches found constructing `KlaySyncingRespResult` with oneOf schemas Boolean, SyncingObject. Input: " + JSON.stringify(instance));
    } else if (match === 0) {
      this.actualInstance = null; // clear the actual instance in case there are multiple matches
      throw new Error("No match found constructing `KlaySyncingRespResult` with oneOf schemas Boolean, SyncingObject. Details: " + errorMessages.join(", "));
    } else {// only 1 match
      // the input is valid
    }
  }

  /**
   * Constructs a <code>KlaySyncingRespResult</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/KlaySyncingRespResult} obj Optional instance to populate.
   * @return {module:model/KlaySyncingRespResult} The populated <code>KlaySyncingRespResult</code> instance.
   */
  _createClass(KlaySyncingRespResult, [{
    key: "getActualInstance",
    value:
    /**
     * Gets the actaul instance, which can be <code>Boolean</code>, <code>SyncingObject</code>.
     * @return {(module:model/Boolean|module:model/SyncingObject)} The actual instance.
     */
    function getActualInstance() {
      return this.actualInstance;
    }

    /**
     * Sets the actaul instance, which can be <code>Boolean</code>, <code>SyncingObject</code>.
     * @param {(module:model/Boolean|module:model/SyncingObject)} obj The actual instance.
     */
  }, {
    key: "setActualInstance",
    value: function setActualInstance(obj) {
      this.actualInstance = KlaySyncingRespResult.constructFromObject(obj).getActualInstance();
    }
  }], [{
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      return new KlaySyncingRespResult(data);
    }
  }]);
  return KlaySyncingRespResult;
}();
/**
 * The block at which the import started (will only be reset, after the sync reached his head).
 * @member {String} startingBlock
 */
_class = KlaySyncingRespResult;
/**
 * Create an instance of KlaySyncingRespResult from a JSON string.
 * @param {string} json_string JSON string.
 * @return {module:model/KlaySyncingRespResult} An instance of KlaySyncingRespResult.
 */
_defineProperty(KlaySyncingRespResult, "fromJSON", function (json_string) {
  return _class.constructFromObject(JSON.parse(json_string));
});
KlaySyncingRespResult.prototype['startingBlock'] = undefined;

/**
 * The current block, same as klay_blockNumber.
 * @member {String} currentBlock
 */
KlaySyncingRespResult.prototype['currentBlock'] = undefined;

/**
 * The estimated highest block.
 * @member {String} highestBlock
 */
KlaySyncingRespResult.prototype['highestBlock'] = undefined;

/**
 * The number of state entries processed until now. If the sync mode is not \"fast\", zero is returned.
 * @member {String} pulledStates
 */
KlaySyncingRespResult.prototype['pulledStates'] = undefined;

/**
 * The number of known state entries that still need to be pulled. If the sync mode is not \"fast\", zero is returned.
 * @member {String} knownStates
 */
KlaySyncingRespResult.prototype['knownStates'] = undefined;
KlaySyncingRespResult.OneOf = ["Boolean", "SyncingObject"];
var _default = exports["default"] = KlaySyncingRespResult;
