// @klaytn/web3rpc/dist/model/KlayGetBlockWithConsensusInfoByNumberRangeRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner = _interopRequireDefault(require("./KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetBlockWithConsensusInfoByNumberRangeRespResult model module.
 * @module model/KlayGetBlockWithConsensusInfoByNumberRangeRespResult
 * @version 0.9.8
 */
var KlayGetBlockWithConsensusInfoByNumberRangeRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResult</code>.
   * Object - A block object with consensus information (a proposer and a list of committee members)
   * @alias module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResult
   */
  function KlayGetBlockWithConsensusInfoByNumberRangeRespResult() {
    _classCallCheck(this, KlayGetBlockWithConsensusInfoByNumberRangeRespResult);
    KlayGetBlockWithConsensusInfoByNumberRangeRespResult.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetBlockWithConsensusInfoByNumberRangeRespResult, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResult} obj Optional instance to populate.
     * @return {module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResult} The populated <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetBlockWithConsensusInfoByNumberRangeRespResult();
        if (data.hasOwnProperty('blockscore')) {
          obj['blockscore'] = _ApiClient["default"].convertToType(data['blockscore'], 'String');
        }
        if (data.hasOwnProperty('committee')) {
          obj['committee'] = _ApiClient["default"].convertToType(data['committee'], ['String']);
        }
        if (data.hasOwnProperty('extraData')) {
          obj['extraData'] = _ApiClient["default"].convertToType(data['extraData'], 'String');
        }
        if (data.hasOwnProperty('gasUsed')) {
          obj['gasUsed'] = _ApiClient["default"].convertToType(data['gasUsed'], 'String');
        }
        if (data.hasOwnProperty('governanceData')) {
          obj['governanceData'] = _ApiClient["default"].convertToType(data['governanceData'], 'String');
        }
        if (data.hasOwnProperty('hash')) {
          obj['hash'] = _ApiClient["default"].convertToType(data['hash'], 'String');
        }
        if (data.hasOwnProperty('logsBloom')) {
          obj['logsBloom'] = _ApiClient["default"].convertToType(data['logsBloom'], 'String');
        }
        if (data.hasOwnProperty('number')) {
          obj['number'] = _ApiClient["default"].convertToType(data['number'], 'String');
        }
        if (data.hasOwnProperty('parentHash')) {
          obj['parentHash'] = _ApiClient["default"].convertToType(data['parentHash'], 'String');
        }
        if (data.hasOwnProperty('proposer')) {
          obj['proposer'] = _ApiClient["default"].convertToType(data['proposer'], 'String');
        }
        if (data.hasOwnProperty('receiptsRoot')) {
          obj['receiptsRoot'] = _ApiClient["default"].convertToType(data['receiptsRoot'], 'String');
        }
        if (data.hasOwnProperty('reward')) {
          obj['reward'] = _ApiClient["default"].convertToType(data['reward'], 'String');
        }
        if (data.hasOwnProperty('size')) {
          obj['size'] = _ApiClient["default"].convertToType(data['size'], 'String');
        }
        if (data.hasOwnProperty('stateRoot')) {
          obj['stateRoot'] = _ApiClient["default"].convertToType(data['stateRoot'], 'String');
        }
        if (data.hasOwnProperty('timestamp')) {
          obj['timestamp'] = _ApiClient["default"].convertToType(data['timestamp'], 'String');
        }
        if (data.hasOwnProperty('timestampFoS')) {
          obj['timestampFoS'] = _ApiClient["default"].convertToType(data['timestampFoS'], 'String');
        }
        if (data.hasOwnProperty('totalBlockScore')) {
          obj['totalBlockScore'] = _ApiClient["default"].convertToType(data['totalBlockScore'], 'String');
        }
        if (data.hasOwnProperty('transactions')) {
          obj['transactions'] = _ApiClient["default"].convertToType(data['transactions'], [_KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner["default"]]);
        }
        if (data.hasOwnProperty('transactionsRoot')) {
          obj['transactionsRoot'] = _ApiClient["default"].convertToType(data['transactionsRoot'], 'String');
        }
        if (data.hasOwnProperty('voteData')) {
          obj['voteData'] = _ApiClient["default"].convertToType(data['voteData'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['blockscore'] && !(typeof data['blockscore'] === 'string' || data['blockscore'] instanceof String)) {
        throw new Error("Expected the field `blockscore` to be a primitive type in the JSON string but got " + data['blockscore']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['committee'])) {
        throw new Error("Expected the field `committee` to be an array in the JSON data but got " + data['committee']);
      }
      // ensure the json data is a string
      if (data['extraData'] && !(typeof data['extraData'] === 'string' || data['extraData'] instanceof String)) {
        throw new Error("Expected the field `extraData` to be a primitive type in the JSON string but got " + data['extraData']);
      }
      // ensure the json data is a string
      if (data['gasUsed'] && !(typeof data['gasUsed'] === 'string' || data['gasUsed'] instanceof String)) {
        throw new Error("Expected the field `gasUsed` to be a primitive type in the JSON string but got " + data['gasUsed']);
      }
      // ensure the json data is a string
      if (data['governanceData'] && !(typeof data['governanceData'] === 'string' || data['governanceData'] instanceof String)) {
        throw new Error("Expected the field `governanceData` to be a primitive type in the JSON string but got " + data['governanceData']);
      }
      // ensure the json data is a string
      if (data['hash'] && !(typeof data['hash'] === 'string' || data['hash'] instanceof String)) {
        throw new Error("Expected the field `hash` to be a primitive type in the JSON string but got " + data['hash']);
      }
      // ensure the json data is a string
      if (data['logsBloom'] && !(typeof data['logsBloom'] === 'string' || data['logsBloom'] instanceof String)) {
        throw new Error("Expected the field `logsBloom` to be a primitive type in the JSON string but got " + data['logsBloom']);
      }
      // ensure the json data is a string
      if (data['number'] && !(typeof data['number'] === 'string' || data['number'] instanceof String)) {
        throw new Error("Expected the field `number` to be a primitive type in the JSON string but got " + data['number']);
      }
      // ensure the json data is a string
      if (data['parentHash'] && !(typeof data['parentHash'] === 'string' || data['parentHash'] instanceof String)) {
        throw new Error("Expected the field `parentHash` to be a primitive type in the JSON string but got " + data['parentHash']);
      }
      // ensure the json data is a string
      if (data['proposer'] && !(typeof data['proposer'] === 'string' || data['proposer'] instanceof String)) {
        throw new Error("Expected the field `proposer` to be a primitive type in the JSON string but got " + data['proposer']);
      }
      // ensure the json data is a string
      if (data['receiptsRoot'] && !(typeof data['receiptsRoot'] === 'string' || data['receiptsRoot'] instanceof String)) {
        throw new Error("Expected the field `receiptsRoot` to be a primitive type in the JSON string but got " + data['receiptsRoot']);
      }
      // ensure the json data is a string
      if (data['reward'] && !(typeof data['reward'] === 'string' || data['reward'] instanceof String)) {
        throw new Error("Expected the field `reward` to be a primitive type in the JSON string but got " + data['reward']);
      }
      // ensure the json data is a string
      if (data['size'] && !(typeof data['size'] === 'string' || data['size'] instanceof String)) {
        throw new Error("Expected the field `size` to be a primitive type in the JSON string but got " + data['size']);
      }
      // ensure the json data is a string
      if (data['stateRoot'] && !(typeof data['stateRoot'] === 'string' || data['stateRoot'] instanceof String)) {
        throw new Error("Expected the field `stateRoot` to be a primitive type in the JSON string but got " + data['stateRoot']);
      }
      // ensure the json data is a string
      if (data['timestamp'] && !(typeof data['timestamp'] === 'string' || data['timestamp'] instanceof String)) {
        throw new Error("Expected the field `timestamp` to be a primitive type in the JSON string but got " + data['timestamp']);
      }
      // ensure the json data is a string
      if (data['timestampFoS'] && !(typeof data['timestampFoS'] === 'string' || data['timestampFoS'] instanceof String)) {
        throw new Error("Expected the field `timestampFoS` to be a primitive type in the JSON string but got " + data['timestampFoS']);
      }
      // ensure the json data is a string
      if (data['totalBlockScore'] && !(typeof data['totalBlockScore'] === 'string' || data['totalBlockScore'] instanceof String)) {
        throw new Error("Expected the field `totalBlockScore` to be a primitive type in the JSON string but got " + data['totalBlockScore']);
      }
      if (data['transactions']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['transactions'])) {
          throw new Error("Expected the field `transactions` to be an array in the JSON data but got " + data['transactions']);
        }
        // validate the optional field `transactions` (array)
        var _iterator = _createForOfIteratorHelper(data['transactions']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['transactionsRoot'] && !(typeof data['transactionsRoot'] === 'string' || data['transactionsRoot'] instanceof String)) {
        throw new Error("Expected the field `transactionsRoot` to be a primitive type in the JSON string but got " + data['transactionsRoot']);
      }
      // ensure the json data is a string
      if (data['voteData'] && !(typeof data['voteData'] === 'string' || data['voteData'] instanceof String)) {
        throw new Error("Expected the field `voteData` to be a primitive type in the JSON string but got " + data['voteData']);
      }
      return true;
    }
  }]);
  return KlayGetBlockWithConsensusInfoByNumberRangeRespResult;
}();
/**
 * @member {String} blockscore
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['blockscore'] = undefined;

/**
 * @member {Array.<String>} committee
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['committee'] = undefined;

/**
 * @member {String} extraData
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['extraData'] = undefined;

/**
 * @member {String} gasUsed
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['gasUsed'] = undefined;

/**
 * @member {String} governanceData
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['governanceData'] = undefined;

/**
 * @member {String} hash
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['hash'] = undefined;

/**
 * @member {String} logsBloom
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['logsBloom'] = undefined;

/**
 * @member {String} number
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['number'] = undefined;

/**
 * @member {String} parentHash
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['parentHash'] = undefined;

/**
 * @member {String} proposer
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['proposer'] = undefined;

/**
 * @member {String} receiptsRoot
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['receiptsRoot'] = undefined;

/**
 * @member {String} reward
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['reward'] = undefined;

/**
 * @member {String} size
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['size'] = undefined;

/**
 * @member {String} stateRoot
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['stateRoot'] = undefined;

/**
 * @member {String} timestamp
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['timestamp'] = undefined;

/**
 * @member {String} timestampFoS
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['timestampFoS'] = undefined;

/**
 * @member {String} totalBlockScore
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['totalBlockScore'] = undefined;

/**
 * @member {Array.<module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner>} transactions
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['transactions'] = undefined;

/**
 * @member {String} transactionsRoot
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['transactionsRoot'] = undefined;

/**
 * @member {String} voteData
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResult.prototype['voteData'] = undefined;
var _default = exports["default"] = KlayGetBlockWithConsensusInfoByNumberRangeRespResult;
