// @klaytn/web3rpc/dist/model/KlayGetChainConfigRespResultGovernanceKip71.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetChainConfigRespResultGovernanceKip71 model module.
 * @module model/KlayGetChainConfigRespResultGovernanceKip71
 * @version 0.9.8
 */
var KlayGetChainConfigRespResultGovernanceKip71 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetChainConfigRespResultGovernanceKip71</code>.
   * @alias module:model/KlayGetChainConfigRespResultGovernanceKip71
   */
  function KlayGetChainConfigRespResultGovernanceKip71() {
    _classCallCheck(this, KlayGetChainConfigRespResultGovernanceKip71);
    KlayGetChainConfigRespResultGovernanceKip71.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetChainConfigRespResultGovernanceKip71, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>KlayGetChainConfigRespResultGovernanceKip71</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetChainConfigRespResultGovernanceKip71} obj Optional instance to populate.
     * @return {module:model/KlayGetChainConfigRespResultGovernanceKip71} The populated <code>KlayGetChainConfigRespResultGovernanceKip71</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetChainConfigRespResultGovernanceKip71();
        if (data.hasOwnProperty('basefeedenominator')) {
          obj['basefeedenominator'] = _ApiClient["default"].convertToType(data['basefeedenominator'], 'Number');
        }
        if (data.hasOwnProperty('gastarget')) {
          obj['gastarget'] = _ApiClient["default"].convertToType(data['gastarget'], 'Number');
        }
        if (data.hasOwnProperty('lowerboundbasefee')) {
          obj['lowerboundbasefee'] = _ApiClient["default"].convertToType(data['lowerboundbasefee'], 'Number');
        }
        if (data.hasOwnProperty('maxblockgasusedforbasefee')) {
          obj['maxblockgasusedforbasefee'] = _ApiClient["default"].convertToType(data['maxblockgasusedforbasefee'], 'Number');
        }
        if (data.hasOwnProperty('upperboundbasefee')) {
          obj['upperboundbasefee'] = _ApiClient["default"].convertToType(data['upperboundbasefee'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetChainConfigRespResultGovernanceKip71</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetChainConfigRespResultGovernanceKip71</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);
  return KlayGetChainConfigRespResultGovernanceKip71;
}();
/**
 * @member {Number} basefeedenominator
 */
KlayGetChainConfigRespResultGovernanceKip71.prototype['basefeedenominator'] = undefined;

/**
 * @member {Number} gastarget
 */
KlayGetChainConfigRespResultGovernanceKip71.prototype['gastarget'] = undefined;

/**
 * @member {Number} lowerboundbasefee
 */
KlayGetChainConfigRespResultGovernanceKip71.prototype['lowerboundbasefee'] = undefined;

/**
 * @member {Number} maxblockgasusedforbasefee
 */
KlayGetChainConfigRespResultGovernanceKip71.prototype['maxblockgasusedforbasefee'] = undefined;

/**
 * @member {Number} upperboundbasefee
 */
KlayGetChainConfigRespResultGovernanceKip71.prototype['upperboundbasefee'] = undefined;
var _default = exports["default"] = KlayGetChainConfigRespResultGovernanceKip71;
