// @klaytn/web3rpc/dist/model/KlayGetAccountKeyRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AccountKeyFail = _interopRequireDefault(require("./AccountKeyFail"));
var _AccountKeyLegacy = _interopRequireDefault(require("./AccountKeyLegacy"));
var _AccountKeyPublic = _interopRequireDefault(require("./AccountKeyPublic"));
var _AccountKeyRoleBased = _interopRequireDefault(require("./AccountKeyRoleBased"));
var _AccountKeyRoleBasedKeyInner = _interopRequireDefault(require("./AccountKeyRoleBasedKeyInner"));
var _AccountKeyWeightedMultiSig = _interopRequireDefault(require("./AccountKeyWeightedMultiSig"));
var _AccountKeyWeightedMultiSigWeightedPublicKeysInner = _interopRequireDefault(require("./AccountKeyWeightedMultiSigWeightedPublicKeysInner"));
var _class;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetAccountKeyRespResult model module.
 * @module model/KlayGetAccountKeyRespResult
 * @version 0.9.8
 */
var KlayGetAccountKeyRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetAccountKeyRespResult</code>.
   * @alias module:model/KlayGetAccountKeyRespResult
   * @param {(module:model/AccountKeyFail|module:model/AccountKeyLegacy|module:model/AccountKeyPublic|module:model/AccountKeyRoleBased|module:model/AccountKeyWeightedMultiSig|module:model/String)} instance The actual instance to initialize KlayGetAccountKeyRespResult.
   */
  function KlayGetAccountKeyRespResult() {
    var instance = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : null;
    _classCallCheck(this, KlayGetAccountKeyRespResult);
    /**
     * Returns the JSON representation of the actual intance.
     * @return {string}
     */
    _defineProperty(this, "toJSON", function () {
      return this.getActualInstance();
    });
    if (instance === null) {
      this.actualInstance = null;
      return;
    }
    var match = 0;
    var errorMessages = [];
    try {
      if (typeof instance === "AccountKeyLegacy") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _AccountKeyLegacy["default"].validateJSON(instance); // throw an exception if no match
        // create AccountKeyLegacy from JS object
        this.actualInstance = _AccountKeyLegacy["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into AccountKeyLegacy
      errorMessages.push("Failed to construct AccountKeyLegacy: " + err);
    }
    try {
      // validate array of string
      if (!(typeof instance === 'string')) {
        throw new Error("Invalid input. Must be string. Input: " + JSON.stringify(instance));
      }
      this.actualInstance = instance;
      match++;
    } catch (err) {
      // json data failed to deserialize into String
      errorMessages.push("Failed to construct String: " + err);
    }
    try {
      if (typeof instance === "AccountKeyPublic") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _AccountKeyPublic["default"].validateJSON(instance); // throw an exception if no match
        // create AccountKeyPublic from JS object
        this.actualInstance = _AccountKeyPublic["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into AccountKeyPublic
      errorMessages.push("Failed to construct AccountKeyPublic: " + err);
    }
    try {
      if (typeof instance === "AccountKeyFail") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _AccountKeyFail["default"].validateJSON(instance); // throw an exception if no match
        // create AccountKeyFail from JS object
        this.actualInstance = _AccountKeyFail["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into AccountKeyFail
      errorMessages.push("Failed to construct AccountKeyFail: " + err);
    }
    try {
      if (typeof instance === "AccountKeyWeightedMultiSig") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _AccountKeyWeightedMultiSig["default"].validateJSON(instance); // throw an exception if no match
        // create AccountKeyWeightedMultiSig from JS object
        this.actualInstance = _AccountKeyWeightedMultiSig["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into AccountKeyWeightedMultiSig
      errorMessages.push("Failed to construct AccountKeyWeightedMultiSig: " + err);
    }
    try {
      if (typeof instance === "AccountKeyRoleBased") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _AccountKeyRoleBased["default"].validateJSON(instance); // throw an exception if no match
        // create AccountKeyRoleBased from JS object
        this.actualInstance = _AccountKeyRoleBased["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into AccountKeyRoleBased
      errorMessages.push("Failed to construct AccountKeyRoleBased: " + err);
    }
    if (match > 1) {
      throw new Error("Multiple matches found constructing `KlayGetAccountKeyRespResult` with oneOf schemas AccountKeyFail, AccountKeyLegacy, AccountKeyPublic, AccountKeyRoleBased, AccountKeyWeightedMultiSig, String. Input: " + JSON.stringify(instance));
    } else if (match === 0) {
      this.actualInstance = null; // clear the actual instance in case there are multiple matches
      throw new Error("No match found constructing `KlayGetAccountKeyRespResult` with oneOf schemas AccountKeyFail, AccountKeyLegacy, AccountKeyPublic, AccountKeyRoleBased, AccountKeyWeightedMultiSig, String. Details: " + errorMessages.join(", "));
    } else {// only 1 match
      // the input is valid
    }
  }

  /**
   * Constructs a <code>KlayGetAccountKeyRespResult</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/KlayGetAccountKeyRespResult} obj Optional instance to populate.
   * @return {module:model/KlayGetAccountKeyRespResult} The populated <code>KlayGetAccountKeyRespResult</code> instance.
   */
  _createClass(KlayGetAccountKeyRespResult, [{
    key: "getActualInstance",
    value:
    /**
     * Gets the actaul instance, which can be <code>AccountKeyFail</code>, <code>AccountKeyLegacy</code>, <code>AccountKeyPublic</code>, <code>AccountKeyRoleBased</code>, <code>AccountKeyWeightedMultiSig</code>, <code>String</code>.
     * @return {(module:model/AccountKeyFail|module:model/AccountKeyLegacy|module:model/AccountKeyPublic|module:model/AccountKeyRoleBased|module:model/AccountKeyWeightedMultiSig|module:model/String)} The actual instance.
     */
    function getActualInstance() {
      return this.actualInstance;
    }

    /**
     * Sets the actaul instance, which can be <code>AccountKeyFail</code>, <code>AccountKeyLegacy</code>, <code>AccountKeyPublic</code>, <code>AccountKeyRoleBased</code>, <code>AccountKeyWeightedMultiSig</code>, <code>String</code>.
     * @param {(module:model/AccountKeyFail|module:model/AccountKeyLegacy|module:model/AccountKeyPublic|module:model/AccountKeyRoleBased|module:model/AccountKeyWeightedMultiSig|module:model/String)} obj The actual instance.
     */
  }, {
    key: "setActualInstance",
    value: function setActualInstance(obj) {
      this.actualInstance = KlayGetAccountKeyRespResult.constructFromObject(obj).getActualInstance();
    }
  }], [{
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      return new KlayGetAccountKeyRespResult(data);
    }
  }]);
  return KlayGetAccountKeyRespResult;
}();
/**
 * A list of keys. A key can be any of AccountKeyNil, AccountKeyLegacy, AccountKeyPublic, AccountKeyFail, and AccountKeyWeightedMultiSig
 * @member {Array.<module:model/AccountKeyRoleBasedKeyInner>} key
 */
_class = KlayGetAccountKeyRespResult;
/**
 * Create an instance of KlayGetAccountKeyRespResult from a JSON string.
 * @param {string} json_string JSON string.
 * @return {module:model/KlayGetAccountKeyRespResult} An instance of KlayGetAccountKeyRespResult.
 */
_defineProperty(KlayGetAccountKeyRespResult, "fromJSON", function (json_string) {
  return _class.constructFromObject(JSON.parse(json_string));
});
KlayGetAccountKeyRespResult.prototype['key'] = undefined;

/**
 * The type of AccountKeyRoleBased. This must be 0x05.
 * @member {Number} keyType
 */
KlayGetAccountKeyRespResult.prototype['keyType'] = undefined;

/**
 * Validation threshold. To be a valid transaction, the weight sum of signatures should be larger than or equal to the threshold.
 * @member {Number} Threshold
 */
KlayGetAccountKeyRespResult.prototype['Threshold'] = undefined;

/**
 * A list of weighted public keys. A weighted public key contains a compressed public key and its weight.
 * @member {Array.<module:model/AccountKeyWeightedMultiSigWeightedPublicKeysInner>} WeightedPublicKeys
 */
KlayGetAccountKeyRespResult.prototype['WeightedPublicKeys'] = undefined;
KlayGetAccountKeyRespResult.OneOf = ["AccountKeyFail", "AccountKeyLegacy", "AccountKeyPublic", "AccountKeyRoleBased", "AccountKeyWeightedMultiSig", "String"];
var _default = exports["default"] = KlayGetAccountKeyRespResult;
