// @klaytn/web3rpc/dist/model/KlayGetAccountRespResultAccountKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _KlayGetAccountRespResultAccountKeyKey = _interopRequireDefault(require("./KlayGetAccountRespResultAccountKeyKey"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetAccountRespResultAccountKey model module.
 * @module model/KlayGetAccountRespResultAccountKey
 * @version 0.9.8
 */
var KlayGetAccountRespResultAccountKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetAccountRespResultAccountKey</code>.
   * @alias module:model/KlayGetAccountRespResultAccountKey
   */
  function KlayGetAccountRespResultAccountKey() {
    _classCallCheck(this, KlayGetAccountRespResultAccountKey);
    KlayGetAccountRespResultAccountKey.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetAccountRespResultAccountKey, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>KlayGetAccountRespResultAccountKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetAccountRespResultAccountKey} obj Optional instance to populate.
     * @return {module:model/KlayGetAccountRespResultAccountKey} The populated <code>KlayGetAccountRespResultAccountKey</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetAccountRespResultAccountKey();
        if (data.hasOwnProperty('key')) {
          obj['key'] = _KlayGetAccountRespResultAccountKeyKey["default"].constructFromObject(data['key']);
        }
        if (data.hasOwnProperty('keyType')) {
          obj['keyType'] = _ApiClient["default"].convertToType(data['keyType'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetAccountRespResultAccountKey</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetAccountRespResultAccountKey</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `key`
      if (data['key']) {
        // data not null
        _KlayGetAccountRespResultAccountKeyKey["default"].validateJSON(data['key']);
      }
      return true;
    }
  }]);
  return KlayGetAccountRespResultAccountKey;
}();
/**
 * @member {module:model/KlayGetAccountRespResultAccountKeyKey} key
 */
KlayGetAccountRespResultAccountKey.prototype['key'] = undefined;

/**
 * @member {Number} keyType
 */
KlayGetAccountRespResultAccountKey.prototype['keyType'] = undefined;
var _default = exports["default"] = KlayGetAccountRespResultAccountKey;
