// @klaytn/web3rpc/dist/model/KlayGetRewardsRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetRewardsRespResult model module.
 * @module model/KlayGetRewardsRespResult
 * @version 0.9.8
 */
var KlayGetRewardsRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetRewardsRespResult</code>.
   * @alias module:model/KlayGetRewardsRespResult
   * @param burntFee {Number} the amount burnt
   * @param kgf {Number} the amount allocated to KGF
   * @param kir {Number} the amount allocated to KIR
   * @param minted {Number} the amount newly minted
   * @param proposer {Number} the amount allocated to the block proposer
   * @param rewards {Object.<String, Number>} mapping from reward recipient to amounts
   * @param stakers {Number} total amount allocated to stakers
   * @param totalFee {Number} total tx fee spent
   */
  function KlayGetRewardsRespResult(burntFee, kgf, kir, minted, proposer, rewards, stakers, totalFee) {
    _classCallCheck(this, KlayGetRewardsRespResult);
    KlayGetRewardsRespResult.initialize(this, burntFee, kgf, kir, minted, proposer, rewards, stakers, totalFee);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetRewardsRespResult, null, [{
    key: "initialize",
    value: function initialize(obj, burntFee, kgf, kir, minted, proposer, rewards, stakers, totalFee) {
      obj['burntFee'] = burntFee;
      obj['kgf'] = kgf;
      obj['kir'] = kir;
      obj['minted'] = minted;
      obj['proposer'] = proposer;
      obj['rewards'] = rewards;
      obj['stakers'] = stakers;
      obj['totalFee'] = totalFee;
    }

    /**
     * Constructs a <code>KlayGetRewardsRespResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetRewardsRespResult} obj Optional instance to populate.
     * @return {module:model/KlayGetRewardsRespResult} The populated <code>KlayGetRewardsRespResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetRewardsRespResult();
        if (data.hasOwnProperty('burntFee')) {
          obj['burntFee'] = _ApiClient["default"].convertToType(data['burntFee'], 'Number');
        }
        if (data.hasOwnProperty('kgf')) {
          obj['kgf'] = _ApiClient["default"].convertToType(data['kgf'], 'Number');
        }
        if (data.hasOwnProperty('kir')) {
          obj['kir'] = _ApiClient["default"].convertToType(data['kir'], 'Number');
        }
        if (data.hasOwnProperty('minted')) {
          obj['minted'] = _ApiClient["default"].convertToType(data['minted'], 'Number');
        }
        if (data.hasOwnProperty('proposer')) {
          obj['proposer'] = _ApiClient["default"].convertToType(data['proposer'], 'Number');
        }
        if (data.hasOwnProperty('rewards')) {
          obj['rewards'] = _ApiClient["default"].convertToType(data['rewards'], {
            'String': 'Number'
          });
        }
        if (data.hasOwnProperty('stakers')) {
          obj['stakers'] = _ApiClient["default"].convertToType(data['stakers'], 'Number');
        }
        if (data.hasOwnProperty('totalFee')) {
          obj['totalFee'] = _ApiClient["default"].convertToType(data['totalFee'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetRewardsRespResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetRewardsRespResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(KlayGetRewardsRespResult.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      return true;
    }
  }]);
  return KlayGetRewardsRespResult;
}();
KlayGetRewardsRespResult.RequiredProperties = ["burntFee", "kgf", "kir", "minted", "proposer", "rewards", "stakers", "totalFee"];

/**
 * the amount burnt
 * @member {Number} burntFee
 */
KlayGetRewardsRespResult.prototype['burntFee'] = undefined;

/**
 * the amount allocated to KGF
 * @member {Number} kgf
 */
KlayGetRewardsRespResult.prototype['kgf'] = undefined;

/**
 * the amount allocated to KIR
 * @member {Number} kir
 */
KlayGetRewardsRespResult.prototype['kir'] = undefined;

/**
 * the amount newly minted
 * @member {Number} minted
 */
KlayGetRewardsRespResult.prototype['minted'] = undefined;

/**
 * the amount allocated to the block proposer
 * @member {Number} proposer
 */
KlayGetRewardsRespResult.prototype['proposer'] = undefined;

/**
 * mapping from reward recipient to amounts
 * @member {Object.<String, Number>} rewards
 */
KlayGetRewardsRespResult.prototype['rewards'] = undefined;

/**
 * total amount allocated to stakers
 * @member {Number} stakers
 */
KlayGetRewardsRespResult.prototype['stakers'] = undefined;

/**
 * total tx fee spent
 * @member {Number} totalFee
 */
KlayGetRewardsRespResult.prototype['totalFee'] = undefined;
var _default = exports["default"] = KlayGetRewardsRespResult;
