// @klaytn/web3rpc/dist/model/KlayGetBlockWithConsensusInfoByNumberResp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _BlockWithConsensusInfo = _interopRequireDefault(require("./BlockWithConsensusInfo"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetBlockWithConsensusInfoByNumberResp model module.
 * @module model/KlayGetBlockWithConsensusInfoByNumberResp
 * @version 0.9.8
 */
var KlayGetBlockWithConsensusInfoByNumberResp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetBlockWithConsensusInfoByNumberResp</code>.
   * @alias module:model/KlayGetBlockWithConsensusInfoByNumberResp
   */
  function KlayGetBlockWithConsensusInfoByNumberResp() {
    _classCallCheck(this, KlayGetBlockWithConsensusInfoByNumberResp);
    KlayGetBlockWithConsensusInfoByNumberResp.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetBlockWithConsensusInfoByNumberResp, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>KlayGetBlockWithConsensusInfoByNumberResp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetBlockWithConsensusInfoByNumberResp} obj Optional instance to populate.
     * @return {module:model/KlayGetBlockWithConsensusInfoByNumberResp} The populated <code>KlayGetBlockWithConsensusInfoByNumberResp</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetBlockWithConsensusInfoByNumberResp();
        if (data.hasOwnProperty('result')) {
          obj['result'] = _BlockWithConsensusInfo["default"].constructFromObject(data['result']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetBlockWithConsensusInfoByNumberResp</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetBlockWithConsensusInfoByNumberResp</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `result`
      if (data['result']) {
        // data not null
        _BlockWithConsensusInfo["default"].validateJSON(data['result']);
      }
      return true;
    }
  }]);
  return KlayGetBlockWithConsensusInfoByNumberResp;
}();
/**
 * @member {module:model/BlockWithConsensusInfo} result
 */
KlayGetBlockWithConsensusInfoByNumberResp.prototype['result'] = undefined;
var _default = exports["default"] = KlayGetBlockWithConsensusInfoByNumberResp;
