// @klaytn/web3rpc/dist/model/KlayGetDecodedAnchoringTransactionByHashRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetDecodedAnchoringTransactionByHashRespResult model module.
 * @module model/KlayGetDecodedAnchoringTransactionByHashRespResult
 * @version 0.9.8
 */
var KlayGetDecodedAnchoringTransactionByHashRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetDecodedAnchoringTransactionByHashRespResult</code>.
   * @alias module:model/KlayGetDecodedAnchoringTransactionByHashRespResult
   * @param blockHash {String} Hash of the child chain block that this anchoring transaction was performed.
   * @param blockNumber {Number} The child chain block number that this anchoring transaction was performed.
   * @param parentHash {String} Hash of the parent block.
   * @param txHash {String} The root of the transaction trie of the block.
   * @param stateRootHash {String} The root of the final state trie of the block.
   * @param receiptHash {String} The root of the receipts trie of the block.
   * @param blockCount {Number} The number of blocks generated during this anchoring period. In most cases, this number is equal to the child chain's SC_TX_PERIOD, with the exception of the case that this transaction was the first anchoring tx after turning on the anchoring.
   * @param txCount {Number} The number of transactions generated in the child chain during this anchoring period.
   */
  function KlayGetDecodedAnchoringTransactionByHashRespResult(blockHash, blockNumber, parentHash, txHash, stateRootHash, receiptHash, blockCount, txCount) {
    _classCallCheck(this, KlayGetDecodedAnchoringTransactionByHashRespResult);
    KlayGetDecodedAnchoringTransactionByHashRespResult.initialize(this, blockHash, blockNumber, parentHash, txHash, stateRootHash, receiptHash, blockCount, txCount);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetDecodedAnchoringTransactionByHashRespResult, null, [{
    key: "initialize",
    value: function initialize(obj, blockHash, blockNumber, parentHash, txHash, stateRootHash, receiptHash, blockCount, txCount) {
      obj['BlockHash'] = blockHash;
      obj['BlockNumber'] = blockNumber;
      obj['ParentHash'] = parentHash;
      obj['TxHash'] = txHash;
      obj['StateRootHash'] = stateRootHash;
      obj['ReceiptHash'] = receiptHash;
      obj['BlockCount'] = blockCount;
      obj['TxCount'] = txCount;
    }

    /**
     * Constructs a <code>KlayGetDecodedAnchoringTransactionByHashRespResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetDecodedAnchoringTransactionByHashRespResult} obj Optional instance to populate.
     * @return {module:model/KlayGetDecodedAnchoringTransactionByHashRespResult} The populated <code>KlayGetDecodedAnchoringTransactionByHashRespResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetDecodedAnchoringTransactionByHashRespResult();
        if (data.hasOwnProperty('BlockHash')) {
          obj['BlockHash'] = _ApiClient["default"].convertToType(data['BlockHash'], 'String');
        }
        if (data.hasOwnProperty('BlockNumber')) {
          obj['BlockNumber'] = _ApiClient["default"].convertToType(data['BlockNumber'], 'Number');
        }
        if (data.hasOwnProperty('ParentHash')) {
          obj['ParentHash'] = _ApiClient["default"].convertToType(data['ParentHash'], 'String');
        }
        if (data.hasOwnProperty('TxHash')) {
          obj['TxHash'] = _ApiClient["default"].convertToType(data['TxHash'], 'String');
        }
        if (data.hasOwnProperty('StateRootHash')) {
          obj['StateRootHash'] = _ApiClient["default"].convertToType(data['StateRootHash'], 'String');
        }
        if (data.hasOwnProperty('ReceiptHash')) {
          obj['ReceiptHash'] = _ApiClient["default"].convertToType(data['ReceiptHash'], 'String');
        }
        if (data.hasOwnProperty('BlockCount')) {
          obj['BlockCount'] = _ApiClient["default"].convertToType(data['BlockCount'], 'Number');
        }
        if (data.hasOwnProperty('TxCount')) {
          obj['TxCount'] = _ApiClient["default"].convertToType(data['TxCount'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetDecodedAnchoringTransactionByHashRespResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetDecodedAnchoringTransactionByHashRespResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(KlayGetDecodedAnchoringTransactionByHashRespResult.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['BlockHash'] && !(typeof data['BlockHash'] === 'string' || data['BlockHash'] instanceof String)) {
        throw new Error("Expected the field `BlockHash` to be a primitive type in the JSON string but got " + data['BlockHash']);
      }
      // ensure the json data is a string
      if (data['ParentHash'] && !(typeof data['ParentHash'] === 'string' || data['ParentHash'] instanceof String)) {
        throw new Error("Expected the field `ParentHash` to be a primitive type in the JSON string but got " + data['ParentHash']);
      }
      // ensure the json data is a string
      if (data['TxHash'] && !(typeof data['TxHash'] === 'string' || data['TxHash'] instanceof String)) {
        throw new Error("Expected the field `TxHash` to be a primitive type in the JSON string but got " + data['TxHash']);
      }
      // ensure the json data is a string
      if (data['StateRootHash'] && !(typeof data['StateRootHash'] === 'string' || data['StateRootHash'] instanceof String)) {
        throw new Error("Expected the field `StateRootHash` to be a primitive type in the JSON string but got " + data['StateRootHash']);
      }
      // ensure the json data is a string
      if (data['ReceiptHash'] && !(typeof data['ReceiptHash'] === 'string' || data['ReceiptHash'] instanceof String)) {
        throw new Error("Expected the field `ReceiptHash` to be a primitive type in the JSON string but got " + data['ReceiptHash']);
      }
      return true;
    }
  }]);
  return KlayGetDecodedAnchoringTransactionByHashRespResult;
}();
KlayGetDecodedAnchoringTransactionByHashRespResult.RequiredProperties = ["BlockHash", "BlockNumber", "ParentHash", "TxHash", "StateRootHash", "ReceiptHash", "BlockCount", "TxCount"];

/**
 * Hash of the child chain block that this anchoring transaction was performed.
 * @member {String} BlockHash
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['BlockHash'] = undefined;

/**
 * The child chain block number that this anchoring transaction was performed.
 * @member {Number} BlockNumber
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['BlockNumber'] = undefined;

/**
 * Hash of the parent block.
 * @member {String} ParentHash
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['ParentHash'] = undefined;

/**
 * The root of the transaction trie of the block.
 * @member {String} TxHash
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['TxHash'] = undefined;

/**
 * The root of the final state trie of the block.
 * @member {String} StateRootHash
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['StateRootHash'] = undefined;

/**
 * The root of the receipts trie of the block.
 * @member {String} ReceiptHash
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['ReceiptHash'] = undefined;

/**
 * The number of blocks generated during this anchoring period. In most cases, this number is equal to the child chain's SC_TX_PERIOD, with the exception of the case that this transaction was the first anchoring tx after turning on the anchoring.
 * @member {Number} BlockCount
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['BlockCount'] = undefined;

/**
 * The number of transactions generated in the child chain during this anchoring period.
 * @member {Number} TxCount
 */
KlayGetDecodedAnchoringTransactionByHashRespResult.prototype['TxCount'] = undefined;
var _default = exports["default"] = KlayGetDecodedAnchoringTransactionByHashRespResult;
