// @klaytn/web3rpc/dist/model/KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInnerSignaturesInner = _interopRequireDefault(require("./KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInnerSignaturesInner"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner model module.
 * @module model/KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner
 * @version 0.9.8
 */
var KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner</code>.
   * @alias module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner
   */
  function KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner() {
    _classCallCheck(this, KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner);
    KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner} obj Optional instance to populate.
     * @return {module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner} The populated <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner();
        if (data.hasOwnProperty('blockHash')) {
          obj['blockHash'] = _ApiClient["default"].convertToType(data['blockHash'], 'String');
        }
        if (data.hasOwnProperty('blockNumber')) {
          obj['blockNumber'] = _ApiClient["default"].convertToType(data['blockNumber'], 'String');
        }
        if (data.hasOwnProperty('contractAddress')) {
          obj['contractAddress'] = _ApiClient["default"].convertToType(data['contractAddress'], 'String');
        }
        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], 'String');
        }
        if (data.hasOwnProperty('gas')) {
          obj['gas'] = _ApiClient["default"].convertToType(data['gas'], 'String');
        }
        if (data.hasOwnProperty('gasPrice')) {
          obj['gasPrice'] = _ApiClient["default"].convertToType(data['gasPrice'], 'String');
        }
        if (data.hasOwnProperty('gasUsed')) {
          obj['gasUsed'] = _ApiClient["default"].convertToType(data['gasUsed'], 'String');
        }
        if (data.hasOwnProperty('logs')) {
          obj['logs'] = _ApiClient["default"].convertToType(data['logs'], [Object]);
        }
        if (data.hasOwnProperty('logsBloom')) {
          obj['logsBloom'] = _ApiClient["default"].convertToType(data['logsBloom'], 'String');
        }
        if (data.hasOwnProperty('nonce')) {
          obj['nonce'] = _ApiClient["default"].convertToType(data['nonce'], 'String');
        }
        if (data.hasOwnProperty('senderTxHash')) {
          obj['senderTxHash'] = _ApiClient["default"].convertToType(data['senderTxHash'], 'String');
        }
        if (data.hasOwnProperty('signatures')) {
          obj['signatures'] = _ApiClient["default"].convertToType(data['signatures'], [_KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInnerSignaturesInner["default"]]);
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }
        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], 'String');
        }
        if (data.hasOwnProperty('transactionHash')) {
          obj['transactionHash'] = _ApiClient["default"].convertToType(data['transactionHash'], 'String');
        }
        if (data.hasOwnProperty('transactionIndex')) {
          obj['transactionIndex'] = _ApiClient["default"].convertToType(data['transactionIndex'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('typeInt')) {
          obj['typeInt'] = _ApiClient["default"].convertToType(data['typeInt'], 'Number');
        }
        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['blockHash'] && !(typeof data['blockHash'] === 'string' || data['blockHash'] instanceof String)) {
        throw new Error("Expected the field `blockHash` to be a primitive type in the JSON string but got " + data['blockHash']);
      }
      // ensure the json data is a string
      if (data['blockNumber'] && !(typeof data['blockNumber'] === 'string' || data['blockNumber'] instanceof String)) {
        throw new Error("Expected the field `blockNumber` to be a primitive type in the JSON string but got " + data['blockNumber']);
      }
      // ensure the json data is a string
      if (data['contractAddress'] && !(typeof data['contractAddress'] === 'string' || data['contractAddress'] instanceof String)) {
        throw new Error("Expected the field `contractAddress` to be a primitive type in the JSON string but got " + data['contractAddress']);
      }
      // ensure the json data is a string
      if (data['from'] && !(typeof data['from'] === 'string' || data['from'] instanceof String)) {
        throw new Error("Expected the field `from` to be a primitive type in the JSON string but got " + data['from']);
      }
      // ensure the json data is a string
      if (data['gas'] && !(typeof data['gas'] === 'string' || data['gas'] instanceof String)) {
        throw new Error("Expected the field `gas` to be a primitive type in the JSON string but got " + data['gas']);
      }
      // ensure the json data is a string
      if (data['gasPrice'] && !(typeof data['gasPrice'] === 'string' || data['gasPrice'] instanceof String)) {
        throw new Error("Expected the field `gasPrice` to be a primitive type in the JSON string but got " + data['gasPrice']);
      }
      // ensure the json data is a string
      if (data['gasUsed'] && !(typeof data['gasUsed'] === 'string' || data['gasUsed'] instanceof String)) {
        throw new Error("Expected the field `gasUsed` to be a primitive type in the JSON string but got " + data['gasUsed']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['logs'])) {
        throw new Error("Expected the field `logs` to be an array in the JSON data but got " + data['logs']);
      }
      // ensure the json data is a string
      if (data['logsBloom'] && !(typeof data['logsBloom'] === 'string' || data['logsBloom'] instanceof String)) {
        throw new Error("Expected the field `logsBloom` to be a primitive type in the JSON string but got " + data['logsBloom']);
      }
      // ensure the json data is a string
      if (data['nonce'] && !(typeof data['nonce'] === 'string' || data['nonce'] instanceof String)) {
        throw new Error("Expected the field `nonce` to be a primitive type in the JSON string but got " + data['nonce']);
      }
      // ensure the json data is a string
      if (data['senderTxHash'] && !(typeof data['senderTxHash'] === 'string' || data['senderTxHash'] instanceof String)) {
        throw new Error("Expected the field `senderTxHash` to be a primitive type in the JSON string but got " + data['senderTxHash']);
      }
      if (data['signatures']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['signatures'])) {
          throw new Error("Expected the field `signatures` to be an array in the JSON data but got " + data['signatures']);
        }
        // validate the optional field `signatures` (array)
        var _iterator = _createForOfIteratorHelper(data['signatures']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInnerSignaturesInner["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['status'] && !(typeof data['status'] === 'string' || data['status'] instanceof String)) {
        throw new Error("Expected the field `status` to be a primitive type in the JSON string but got " + data['status']);
      }
      // ensure the json data is a string
      if (data['to'] && !(typeof data['to'] === 'string' || data['to'] instanceof String)) {
        throw new Error("Expected the field `to` to be a primitive type in the JSON string but got " + data['to']);
      }
      // ensure the json data is a string
      if (data['transactionHash'] && !(typeof data['transactionHash'] === 'string' || data['transactionHash'] instanceof String)) {
        throw new Error("Expected the field `transactionHash` to be a primitive type in the JSON string but got " + data['transactionHash']);
      }
      // ensure the json data is a string
      if (data['transactionIndex'] && !(typeof data['transactionIndex'] === 'string' || data['transactionIndex'] instanceof String)) {
        throw new Error("Expected the field `transactionIndex` to be a primitive type in the JSON string but got " + data['transactionIndex']);
      }
      // ensure the json data is a string
      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }
      // ensure the json data is a string
      if (data['value'] && !(typeof data['value'] === 'string' || data['value'] instanceof String)) {
        throw new Error("Expected the field `value` to be a primitive type in the JSON string but got " + data['value']);
      }
      return true;
    }
  }]);
  return KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner;
}();
/**
 * @member {String} blockHash
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['blockHash'] = undefined;

/**
 * @member {String} blockNumber
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['blockNumber'] = undefined;

/**
 * @member {String} contractAddress
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['contractAddress'] = undefined;

/**
 * @member {String} from
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['from'] = undefined;

/**
 * @member {String} gas
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['gas'] = undefined;

/**
 * @member {String} gasPrice
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['gasPrice'] = undefined;

/**
 * @member {String} gasUsed
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['gasUsed'] = undefined;

/**
 * @member {Array.<Object>} logs
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['logs'] = undefined;

/**
 * @member {String} logsBloom
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['logsBloom'] = undefined;

/**
 * @member {String} nonce
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['nonce'] = undefined;

/**
 * @member {String} senderTxHash
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['senderTxHash'] = undefined;

/**
 * @member {Array.<module:model/KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInnerSignaturesInner>} signatures
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['signatures'] = undefined;

/**
 * @member {String} status
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['status'] = undefined;

/**
 * @member {String} to
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['to'] = undefined;

/**
 * @member {String} transactionHash
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['transactionHash'] = undefined;

/**
 * @member {String} transactionIndex
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['transactionIndex'] = undefined;

/**
 * @member {String} type
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['type'] = undefined;

/**
 * @member {Number} typeInt
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['typeInt'] = undefined;

/**
 * @member {String} value
 */
KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner.prototype['value'] = undefined;
var _default = exports["default"] = KlayGetBlockWithConsensusInfoByNumberRangeRespResultTransactionsInner;
