// @klaytn/web3rpc/dist/model/MemStatsRespResultBySizeInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The MemStatsRespResultBySizeInner model module.
 * @module model/MemStatsRespResultBySizeInner
 * @version 0.9.8
 */
var MemStatsRespResultBySizeInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemStatsRespResultBySizeInner</code>.
   * @alias module:model/MemStatsRespResultBySizeInner
   */
  function MemStatsRespResultBySizeInner() {
    _classCallCheck(this, MemStatsRespResultBySizeInner);
    MemStatsRespResultBySizeInner.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(MemStatsRespResultBySizeInner, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>MemStatsRespResultBySizeInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemStatsRespResultBySizeInner} obj Optional instance to populate.
     * @return {module:model/MemStatsRespResultBySizeInner} The populated <code>MemStatsRespResultBySizeInner</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemStatsRespResultBySizeInner();
        if (data.hasOwnProperty('Size')) {
          obj['Size'] = _ApiClient["default"].convertToType(data['Size'], 'String');
        }
        if (data.hasOwnProperty('Mallocs')) {
          obj['Mallocs'] = _ApiClient["default"].convertToType(data['Mallocs'], 'String');
        }
        if (data.hasOwnProperty('Frees')) {
          obj['Frees'] = _ApiClient["default"].convertToType(data['Frees'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>MemStatsRespResultBySizeInner</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>MemStatsRespResultBySizeInner</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['Size'] && !(typeof data['Size'] === 'string' || data['Size'] instanceof String)) {
        throw new Error("Expected the field `Size` to be a primitive type in the JSON string but got " + data['Size']);
      }
      // ensure the json data is a string
      if (data['Mallocs'] && !(typeof data['Mallocs'] === 'string' || data['Mallocs'] instanceof String)) {
        throw new Error("Expected the field `Mallocs` to be a primitive type in the JSON string but got " + data['Mallocs']);
      }
      // ensure the json data is a string
      if (data['Frees'] && !(typeof data['Frees'] === 'string' || data['Frees'] instanceof String)) {
        throw new Error("Expected the field `Frees` to be a primitive type in the JSON string but got " + data['Frees']);
      }
      return true;
    }
  }]);
  return MemStatsRespResultBySizeInner;
}();
/**
 * Size is the maximum byte size of an object in this size class.
 * @member {String} Size
 */
MemStatsRespResultBySizeInner.prototype['Size'] = undefined;

/**
 * Mallocs is the cumulative count of heap objects allocated in this size class. The cumulative bytes of allocation is Size*Mallocs. The number of live objects in this size class is Mallocs - Frees.
 * @member {String} Mallocs
 */
MemStatsRespResultBySizeInner.prototype['Mallocs'] = undefined;

/**
 * Frees is the cumulative count of heap objects freed in this size class.
 * @member {String} Frees
 */
MemStatsRespResultBySizeInner.prototype['Frees'] = undefined;
var _default = exports["default"] = MemStatsRespResultBySizeInner;
