// @klaytn/web3rpc/dist/model/MyVoteList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The MyVoteList model module.
 * @module model/MyVoteList
 * @version 0.9.8
 */
var MyVoteList = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MyVoteList</code>.
   * @alias module:model/MyVoteList
   * @param blockNum {Number} The block number that this vote is stored
   * @param casted {Boolean} If this vote is stored in a block or not
   * @param key {String} The content of the vote
   * @param value {String} The content of the vote
   */
  function MyVoteList(blockNum, casted, key, value) {
    _classCallCheck(this, MyVoteList);
    MyVoteList.initialize(this, blockNum, casted, key, value);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(MyVoteList, null, [{
    key: "initialize",
    value: function initialize(obj, blockNum, casted, key, value) {
      obj['BlockNum'] = blockNum;
      obj['Casted'] = casted;
      obj['Key'] = key;
      obj['Value'] = value;
    }

    /**
     * Constructs a <code>MyVoteList</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MyVoteList} obj Optional instance to populate.
     * @return {module:model/MyVoteList} The populated <code>MyVoteList</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MyVoteList();
        if (data.hasOwnProperty('BlockNum')) {
          obj['BlockNum'] = _ApiClient["default"].convertToType(data['BlockNum'], 'Number');
        }
        if (data.hasOwnProperty('Casted')) {
          obj['Casted'] = _ApiClient["default"].convertToType(data['Casted'], 'Boolean');
        }
        if (data.hasOwnProperty('Key')) {
          obj['Key'] = _ApiClient["default"].convertToType(data['Key'], 'String');
        }
        if (data.hasOwnProperty('Value')) {
          obj['Value'] = _ApiClient["default"].convertToType(data['Value'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>MyVoteList</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>MyVoteList</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(MyVoteList.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['Key'] && !(typeof data['Key'] === 'string' || data['Key'] instanceof String)) {
        throw new Error("Expected the field `Key` to be a primitive type in the JSON string but got " + data['Key']);
      }
      // ensure the json data is a string
      if (data['Value'] && !(typeof data['Value'] === 'string' || data['Value'] instanceof String)) {
        throw new Error("Expected the field `Value` to be a primitive type in the JSON string but got " + data['Value']);
      }
      return true;
    }
  }]);
  return MyVoteList;
}();
MyVoteList.RequiredProperties = ["BlockNum", "Casted", "Key", "Value"];

/**
 * The block number that this vote is stored
 * @member {Number} BlockNum
 */
MyVoteList.prototype['BlockNum'] = undefined;

/**
 * If this vote is stored in a block or not
 * @member {Boolean} Casted
 */
MyVoteList.prototype['Casted'] = undefined;

/**
 * The content of the vote
 * @member {String} Key
 */
MyVoteList.prototype['Key'] = undefined;

/**
 * The content of the vote
 * @member {String} Value
 */
MyVoteList.prototype['Value'] = undefined;
var _default = exports["default"] = MyVoteList;
