// @klaytn/web3rpc/dist/model/SendArgs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _SendArgsAccessListInner = _interopRequireDefault(require("./SendArgsAccessListInner"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The SendArgs model module.
 * @module model/SendArgs
 * @version 0.9.8
 */
var SendArgs = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SendArgs</code>.
   * @alias module:model/SendArgs
   * @param nonce {String} 
   * @param to {String} 
   */
  function SendArgs(nonce, to) {
    _classCallCheck(this, SendArgs);
    SendArgs.initialize(this, nonce, to);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(SendArgs, null, [{
    key: "initialize",
    value: function initialize(obj, nonce, to) {
      obj['nonce'] = nonce;
      obj['to'] = to;
    }

    /**
     * Constructs a <code>SendArgs</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SendArgs} obj Optional instance to populate.
     * @return {module:model/SendArgs} The populated <code>SendArgs</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SendArgs();
        if (data.hasOwnProperty('accessList')) {
          obj['accessList'] = _ApiClient["default"].convertToType(data['accessList'], [_SendArgsAccessListInner["default"]]);
        }
        if (data.hasOwnProperty('chainId')) {
          obj['chainId'] = _ApiClient["default"].convertToType(data['chainId'], 'String');
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }
        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], 'String');
        }
        if (data.hasOwnProperty('gas')) {
          obj['gas'] = _ApiClient["default"].convertToType(data['gas'], 'String');
        }
        if (data.hasOwnProperty('gasPrice')) {
          obj['gasPrice'] = _ApiClient["default"].convertToType(data['gasPrice'], 'String');
        }
        if (data.hasOwnProperty('input')) {
          obj['input'] = _ApiClient["default"].convertToType(data['input'], 'String');
        }
        if (data.hasOwnProperty('maxFeePerGas')) {
          obj['maxFeePerGas'] = _ApiClient["default"].convertToType(data['maxFeePerGas'], 'String');
        }
        if (data.hasOwnProperty('maxPriorityFeePerGas')) {
          obj['maxPriorityFeePerGas'] = _ApiClient["default"].convertToType(data['maxPriorityFeePerGas'], 'String');
        }
        if (data.hasOwnProperty('nonce')) {
          obj['nonce'] = _ApiClient["default"].convertToType(data['nonce'], 'String');
        }
        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], 'String');
        }
        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>SendArgs</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SendArgs</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(SendArgs.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['accessList']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accessList'])) {
          throw new Error("Expected the field `accessList` to be an array in the JSON data but got " + data['accessList']);
        }
        // validate the optional field `accessList` (array)
        var _iterator2 = _createForOfIteratorHelper(data['accessList']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _SendArgsAccessListInner["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['chainId'] && !(typeof data['chainId'] === 'string' || data['chainId'] instanceof String)) {
        throw new Error("Expected the field `chainId` to be a primitive type in the JSON string but got " + data['chainId']);
      }
      // ensure the json data is a string
      if (data['data'] && !(typeof data['data'] === 'string' || data['data'] instanceof String)) {
        throw new Error("Expected the field `data` to be a primitive type in the JSON string but got " + data['data']);
      }
      // ensure the json data is a string
      if (data['from'] && !(typeof data['from'] === 'string' || data['from'] instanceof String)) {
        throw new Error("Expected the field `from` to be a primitive type in the JSON string but got " + data['from']);
      }
      // ensure the json data is a string
      if (data['gas'] && !(typeof data['gas'] === 'string' || data['gas'] instanceof String)) {
        throw new Error("Expected the field `gas` to be a primitive type in the JSON string but got " + data['gas']);
      }
      // ensure the json data is a string
      if (data['gasPrice'] && !(typeof data['gasPrice'] === 'string' || data['gasPrice'] instanceof String)) {
        throw new Error("Expected the field `gasPrice` to be a primitive type in the JSON string but got " + data['gasPrice']);
      }
      // ensure the json data is a string
      if (data['input'] && !(typeof data['input'] === 'string' || data['input'] instanceof String)) {
        throw new Error("Expected the field `input` to be a primitive type in the JSON string but got " + data['input']);
      }
      // ensure the json data is a string
      if (data['maxFeePerGas'] && !(typeof data['maxFeePerGas'] === 'string' || data['maxFeePerGas'] instanceof String)) {
        throw new Error("Expected the field `maxFeePerGas` to be a primitive type in the JSON string but got " + data['maxFeePerGas']);
      }
      // ensure the json data is a string
      if (data['maxPriorityFeePerGas'] && !(typeof data['maxPriorityFeePerGas'] === 'string' || data['maxPriorityFeePerGas'] instanceof String)) {
        throw new Error("Expected the field `maxPriorityFeePerGas` to be a primitive type in the JSON string but got " + data['maxPriorityFeePerGas']);
      }
      // ensure the json data is a string
      if (data['nonce'] && !(typeof data['nonce'] === 'string' || data['nonce'] instanceof String)) {
        throw new Error("Expected the field `nonce` to be a primitive type in the JSON string but got " + data['nonce']);
      }
      // ensure the json data is a string
      if (data['to'] && !(typeof data['to'] === 'string' || data['to'] instanceof String)) {
        throw new Error("Expected the field `to` to be a primitive type in the JSON string but got " + data['to']);
      }
      // ensure the json data is a string
      if (data['value'] && !(typeof data['value'] === 'string' || data['value'] instanceof String)) {
        throw new Error("Expected the field `value` to be a primitive type in the JSON string but got " + data['value']);
      }
      return true;
    }
  }]);
  return SendArgs;
}();
SendArgs.RequiredProperties = ["nonce", "to"];

/**
 * @member {Array.<module:model/SendArgsAccessListInner>} accessList
 */
SendArgs.prototype['accessList'] = undefined;

/**
 * @member {String} chainId
 */
SendArgs.prototype['chainId'] = undefined;

/**
 * @member {String} data
 */
SendArgs.prototype['data'] = undefined;

/**
 * @member {String} from
 */
SendArgs.prototype['from'] = undefined;

/**
 * @member {String} gas
 */
SendArgs.prototype['gas'] = undefined;

/**
 * @member {String} gasPrice
 */
SendArgs.prototype['gasPrice'] = undefined;

/**
 * @member {String} input
 */
SendArgs.prototype['input'] = undefined;

/**
 * @member {String} maxFeePerGas
 */
SendArgs.prototype['maxFeePerGas'] = undefined;

/**
 * @member {String} maxPriorityFeePerGas
 */
SendArgs.prototype['maxPriorityFeePerGas'] = undefined;

/**
 * @member {String} nonce
 */
SendArgs.prototype['nonce'] = undefined;

/**
 * @member {String} to
 */
SendArgs.prototype['to'] = undefined;

/**
 * @member {String} value
 */
SendArgs.prototype['value'] = undefined;
var _default = exports["default"] = SendArgs;
