// @klaytn/web3rpc/dist/model/SpamThrottlerConfigRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The SpamThrottlerConfigRespResult model module.
 * @module model/SpamThrottlerConfigRespResult
 * @version 0.9.8
 */
var SpamThrottlerConfigRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SpamThrottlerConfigRespResult</code>.
   * @alias module:model/SpamThrottlerConfigRespResult
   * @param activateTxPoolSize {Number} 
   * @param targetFailRatio {Number} 
   * @param throttleTps {Number} 
   * @param maxCandidates {Number} 
   * @param increaseWeight {Number} 
   * @param decreaseWeight {Number} 
   * @param initialThreshold {Number} 
   * @param minimumThreshold {Number} 
   * @param thresholdAdjustment {Number} 
   * @param throttleSeconds {Number} 
   */
  function SpamThrottlerConfigRespResult(activateTxPoolSize, targetFailRatio, throttleTps, maxCandidates, increaseWeight, decreaseWeight, initialThreshold, minimumThreshold, thresholdAdjustment, throttleSeconds) {
    _classCallCheck(this, SpamThrottlerConfigRespResult);
    SpamThrottlerConfigRespResult.initialize(this, activateTxPoolSize, targetFailRatio, throttleTps, maxCandidates, increaseWeight, decreaseWeight, initialThreshold, minimumThreshold, thresholdAdjustment, throttleSeconds);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(SpamThrottlerConfigRespResult, null, [{
    key: "initialize",
    value: function initialize(obj, activateTxPoolSize, targetFailRatio, throttleTps, maxCandidates, increaseWeight, decreaseWeight, initialThreshold, minimumThreshold, thresholdAdjustment, throttleSeconds) {
      obj['activate_tx_pool_size'] = activateTxPoolSize;
      obj['target_fail_ratio'] = targetFailRatio;
      obj['throttle_tps'] = throttleTps;
      obj['max_candidates'] = maxCandidates;
      obj['increase_weight'] = increaseWeight;
      obj['decrease_weight'] = decreaseWeight;
      obj['initial_threshold'] = initialThreshold;
      obj['minimum_threshold'] = minimumThreshold;
      obj['threshold_adjustment'] = thresholdAdjustment;
      obj['throttle_seconds'] = throttleSeconds;
    }

    /**
     * Constructs a <code>SpamThrottlerConfigRespResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SpamThrottlerConfigRespResult} obj Optional instance to populate.
     * @return {module:model/SpamThrottlerConfigRespResult} The populated <code>SpamThrottlerConfigRespResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SpamThrottlerConfigRespResult();
        if (data.hasOwnProperty('activate_tx_pool_size')) {
          obj['activate_tx_pool_size'] = _ApiClient["default"].convertToType(data['activate_tx_pool_size'], 'Number');
        }
        if (data.hasOwnProperty('target_fail_ratio')) {
          obj['target_fail_ratio'] = _ApiClient["default"].convertToType(data['target_fail_ratio'], 'Number');
        }
        if (data.hasOwnProperty('throttle_tps')) {
          obj['throttle_tps'] = _ApiClient["default"].convertToType(data['throttle_tps'], 'Number');
        }
        if (data.hasOwnProperty('max_candidates')) {
          obj['max_candidates'] = _ApiClient["default"].convertToType(data['max_candidates'], 'Number');
        }
        if (data.hasOwnProperty('increase_weight')) {
          obj['increase_weight'] = _ApiClient["default"].convertToType(data['increase_weight'], 'Number');
        }
        if (data.hasOwnProperty('decrease_weight')) {
          obj['decrease_weight'] = _ApiClient["default"].convertToType(data['decrease_weight'], 'Number');
        }
        if (data.hasOwnProperty('initial_threshold')) {
          obj['initial_threshold'] = _ApiClient["default"].convertToType(data['initial_threshold'], 'Number');
        }
        if (data.hasOwnProperty('minimum_threshold')) {
          obj['minimum_threshold'] = _ApiClient["default"].convertToType(data['minimum_threshold'], 'Number');
        }
        if (data.hasOwnProperty('threshold_adjustment')) {
          obj['threshold_adjustment'] = _ApiClient["default"].convertToType(data['threshold_adjustment'], 'Number');
        }
        if (data.hasOwnProperty('throttle_seconds')) {
          obj['throttle_seconds'] = _ApiClient["default"].convertToType(data['throttle_seconds'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>SpamThrottlerConfigRespResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SpamThrottlerConfigRespResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(SpamThrottlerConfigRespResult.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      return true;
    }
  }]);
  return SpamThrottlerConfigRespResult;
}();
SpamThrottlerConfigRespResult.RequiredProperties = ["activate_tx_pool_size", "target_fail_ratio", "throttle_tps", "max_candidates", "increase_weight", "decrease_weight", "initial_threshold", "minimum_threshold", "threshold_adjustment", "throttle_seconds"];

/**
 * @member {Number} activate_tx_pool_size
 */
SpamThrottlerConfigRespResult.prototype['activate_tx_pool_size'] = undefined;

/**
 * @member {Number} target_fail_ratio
 */
SpamThrottlerConfigRespResult.prototype['target_fail_ratio'] = undefined;

/**
 * @member {Number} throttle_tps
 */
SpamThrottlerConfigRespResult.prototype['throttle_tps'] = undefined;

/**
 * @member {Number} max_candidates
 */
SpamThrottlerConfigRespResult.prototype['max_candidates'] = undefined;

/**
 * @member {Number} increase_weight
 */
SpamThrottlerConfigRespResult.prototype['increase_weight'] = undefined;

/**
 * @member {Number} decrease_weight
 */
SpamThrottlerConfigRespResult.prototype['decrease_weight'] = undefined;

/**
 * @member {Number} initial_threshold
 */
SpamThrottlerConfigRespResult.prototype['initial_threshold'] = undefined;

/**
 * @member {Number} minimum_threshold
 */
SpamThrottlerConfigRespResult.prototype['minimum_threshold'] = undefined;

/**
 * @member {Number} threshold_adjustment
 */
SpamThrottlerConfigRespResult.prototype['threshold_adjustment'] = undefined;

/**
 * @member {Number} throttle_seconds
 */
SpamThrottlerConfigRespResult.prototype['throttle_seconds'] = undefined;
var _default = exports["default"] = SpamThrottlerConfigRespResult;
