// @klaytn/web3rpc/dist/model/StandardTracingOptions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The StandardTracingOptions model module.
 * @module model/StandardTracingOptions
 * @version 0.9.8
 */
var StandardTracingOptions = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>StandardTracingOptions</code>.
   * You may give trace API function a secondary optional argument, which specifies the options for this specific call.
   * @alias module:model/StandardTracingOptions
   */
  function StandardTracingOptions() {
    _classCallCheck(this, StandardTracingOptions);
    StandardTracingOptions.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(StandardTracingOptions, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>StandardTracingOptions</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/StandardTracingOptions} obj Optional instance to populate.
     * @return {module:model/StandardTracingOptions} The populated <code>StandardTracingOptions</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new StandardTracingOptions();
        if (data.hasOwnProperty('disableStorage')) {
          obj['disableStorage'] = _ApiClient["default"].convertToType(data['disableStorage'], 'Boolean');
        }
        if (data.hasOwnProperty('disableMemory')) {
          obj['disableMemory'] = _ApiClient["default"].convertToType(data['disableMemory'], 'Boolean');
        }
        if (data.hasOwnProperty('disableStack')) {
          obj['disableStack'] = _ApiClient["default"].convertToType(data['disableStack'], 'Boolean');
        }
        if (data.hasOwnProperty('txHash')) {
          obj['txHash'] = _ApiClient["default"].convertToType(data['txHash'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>StandardTracingOptions</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>StandardTracingOptions</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['txHash'] && !(typeof data['txHash'] === 'string' || data['txHash'] instanceof String)) {
        throw new Error("Expected the field `txHash` to be a primitive type in the JSON string but got " + data['txHash']);
      }
      return true;
    }
  }]);
  return StandardTracingOptions;
}();
/**
 * Setting this to true will disable storage capture.
 * @member {Boolean} disableStorage
 * @default false
 */
StandardTracingOptions.prototype['disableStorage'] = false;

/**
 * Setting this to true will disable memory capture.
 * @member {Boolean} disableMemory
 * @default false
 */
StandardTracingOptions.prototype['disableMemory'] = false;

/**
 * Setting this to true will disable stack capture.
 * @member {Boolean} disableStack
 * @default false
 */
StandardTracingOptions.prototype['disableStack'] = false;

/**
 * Setting this value will trace only the specified transaction.
 * @member {String} txHash
 */
StandardTracingOptions.prototype['txHash'] = undefined;
var _default = exports["default"] = StandardTracingOptions;
