// @klaytn/web3rpc/dist/model/StateMigrationStatusRespResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
 * web3rpc
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.9.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The StateMigrationStatusRespResult model module.
 * @module model/StateMigrationStatusRespResult
 * @version 0.9.8
 */
var StateMigrationStatusRespResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>StateMigrationStatusRespResult</code>.
   * Returns the status information of the state migration
   * @alias module:model/StateMigrationStatusRespResult
   * @param committed {Number} committed is the number of trie nodes that have been copied by the state migration.
   * @param err {String} null if the state migration finished well, or an error if not.
   * @param isMigration {Boolean} true if the state migration is running, or false if not.
   * @param migrationBlockNumber {Number} a blockNumber which the state migration started at. (0 if the state migration is not running.)
   * @param pending {Number} pending represents the number of trie nodes that have not been processed by the state migration.
   * @param progress {Number} progress is the progress of the state migration calculated in percent.
   * @param read {Number} read represents the number of trie nodes that have been read by the state migration.
   */
  function StateMigrationStatusRespResult(committed, err, isMigration, migrationBlockNumber, pending, progress, read) {
    _classCallCheck(this, StateMigrationStatusRespResult);
    StateMigrationStatusRespResult.initialize(this, committed, err, isMigration, migrationBlockNumber, pending, progress, read);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(StateMigrationStatusRespResult, null, [{
    key: "initialize",
    value: function initialize(obj, committed, err, isMigration, migrationBlockNumber, pending, progress, read) {
      obj['committed'] = committed;
      obj['err'] = err;
      obj['isMigration'] = isMigration;
      obj['migrationBlockNumber'] = migrationBlockNumber;
      obj['pending'] = pending;
      obj['progress'] = progress;
      obj['read'] = read;
    }

    /**
     * Constructs a <code>StateMigrationStatusRespResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/StateMigrationStatusRespResult} obj Optional instance to populate.
     * @return {module:model/StateMigrationStatusRespResult} The populated <code>StateMigrationStatusRespResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new StateMigrationStatusRespResult();
        if (data.hasOwnProperty('committed')) {
          obj['committed'] = _ApiClient["default"].convertToType(data['committed'], 'Number');
        }
        if (data.hasOwnProperty('err')) {
          obj['err'] = _ApiClient["default"].convertToType(data['err'], 'String');
        }
        if (data.hasOwnProperty('isMigration')) {
          obj['isMigration'] = _ApiClient["default"].convertToType(data['isMigration'], 'Boolean');
        }
        if (data.hasOwnProperty('migrationBlockNumber')) {
          obj['migrationBlockNumber'] = _ApiClient["default"].convertToType(data['migrationBlockNumber'], 'Number');
        }
        if (data.hasOwnProperty('pending')) {
          obj['pending'] = _ApiClient["default"].convertToType(data['pending'], 'Number');
        }
        if (data.hasOwnProperty('progress')) {
          obj['progress'] = _ApiClient["default"].convertToType(data['progress'], 'Number');
        }
        if (data.hasOwnProperty('read')) {
          obj['read'] = _ApiClient["default"].convertToType(data['read'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>StateMigrationStatusRespResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>StateMigrationStatusRespResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(StateMigrationStatusRespResult.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['err'] && !(typeof data['err'] === 'string' || data['err'] instanceof String)) {
        throw new Error("Expected the field `err` to be a primitive type in the JSON string but got " + data['err']);
      }
      return true;
    }
  }]);
  return StateMigrationStatusRespResult;
}();
StateMigrationStatusRespResult.RequiredProperties = ["committed", "err", "isMigration", "migrationBlockNumber", "pending", "progress", "read"];

/**
 * committed is the number of trie nodes that have been copied by the state migration.
 * @member {Number} committed
 */
StateMigrationStatusRespResult.prototype['committed'] = undefined;

/**
 * null if the state migration finished well, or an error if not.
 * @member {String} err
 */
StateMigrationStatusRespResult.prototype['err'] = undefined;

/**
 * true if the state migration is running, or false if not.
 * @member {Boolean} isMigration
 */
StateMigrationStatusRespResult.prototype['isMigration'] = undefined;

/**
 * a blockNumber which the state migration started at. (0 if the state migration is not running.)
 * @member {Number} migrationBlockNumber
 */
StateMigrationStatusRespResult.prototype['migrationBlockNumber'] = undefined;

/**
 * pending represents the number of trie nodes that have not been processed by the state migration.
 * @member {Number} pending
 */
StateMigrationStatusRespResult.prototype['pending'] = undefined;

/**
 * progress is the progress of the state migration calculated in percent.
 * @member {Number} progress
 */
StateMigrationStatusRespResult.prototype['progress'] = undefined;

/**
 * read represents the number of trie nodes that have been read by the state migration.
 * @member {Number} read
 */
StateMigrationStatusRespResult.prototype['read'] = undefined;
var _default = exports["default"] = StateMigrationStatusRespResult;
